// shippo/models/components/address.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Address$inboundSchema = exports.Longitude$inboundSchema = exports.Latitude$inboundSchema = void 0;
exports.latitudeFromJSON = latitudeFromJSON;
exports.longitudeFromJSON = longitudeFromJSON;
exports.addressFromJSON = addressFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addressvalidationresults_js_1 = require("./addressvalidationresults.js");
/** @internal */
exports.Latitude$inboundSchema = z.union([z.number(), z.string()]);
function latitudeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Latitude$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Latitude' from JSON`);
}
/** @internal */
exports.Longitude$inboundSchema = z
    .union([z.number(), z.string()]);
function longitudeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Longitude$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Longitude' from JSON`);
}
/** @internal */
exports.Address$inboundSchema = z.pipe(z.object({
    name: z.optional(z.string()),
    company: z.optional(z.string()),
    street1: z.optional(z.string()),
    street2: z.optional(z.string()),
    street3: z.optional(z.string()),
    street_no: z.optional(z.string()),
    city: z.optional(z.string()),
    state: z.optional(z.string()),
    zip: z.optional(z.string()),
    country: z.string(),
    phone: z.optional(z.string()),
    email: z.optional(z.string()),
    is_residential: z.optional(z.boolean()),
    metadata: z.optional(z.string()),
    is_complete: z.optional(z.boolean()),
    latitude: z.optional(z.union([z.number(), z.string()])),
    longitude: z.optional(z.union([z.number(), z.string()])),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    validation_results: z.optional(addressvalidationresults_js_1.AddressValidationResults$inboundSchema),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "street_no": "streetNo",
        "is_residential": "isResidential",
        "is_complete": "isComplete",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_updated": "objectUpdated",
        "validation_results": "validationResults",
    });
}));
function addressFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Address$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Address' from JSON`);
}
