// shippo/models/components/alcohol.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alcohol$outboundSchema = exports.Alcohol$inboundSchema = exports.RecipientType$outboundSchema = exports.RecipientType$inboundSchema = exports.RecipientType = void 0;
exports.alcoholToJSON = alcoholToJSON;
exports.alcoholFromJSON = alcoholFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Mandatory for Fedex only. License type of the recipient of the Alcohol Package.
 */
exports.RecipientType = {
    Licensee: "licensee",
    Consumer: "consumer",
};
/** @internal */
exports.RecipientType$inboundSchema = z.enum(exports.RecipientType);
/** @internal */
exports.RecipientType$outboundSchema = exports.RecipientType$inboundSchema;
/** @internal */
exports.Alcohol$inboundSchema = z.pipe(z.object({
    contains_alcohol: z.optional(z.boolean()),
    recipient_type: z.optional(exports.RecipientType$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "contains_alcohol": "containsAlcohol",
        "recipient_type": "recipientType",
    });
}));
/** @internal */
exports.Alcohol$outboundSchema = z.pipe(z.object({
    containsAlcohol: z.optional(z.boolean()),
    recipientType: z.optional(exports.RecipientType$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        containsAlcohol: "contains_alcohol",
        recipientType: "recipient_type",
    });
}));
function alcoholToJSON(alcohol) {
    return JSON.stringify(exports.Alcohol$outboundSchema.parse(alcohol));
}
function alcoholFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Alcohol$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Alcohol' from JSON`);
}
