// shippo/models/components/batch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Batch$inboundSchema = exports.BatchStatus$inboundSchema = exports.ObjectResults$inboundSchema = exports.BatchStatus = void 0;
exports.objectResultsFromJSON = objectResultsFromJSON;
exports.batchFromJSON = batchFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const batchshipmentpaginatedlist_js_1 = require("./batchshipmentpaginatedlist.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
/**
 * Batches that are `VALIDATING` are being created and validated<br>
 *
 * @remarks
 * `VALID` batches can be purchased<br>
 * `INVALID` batches cannot be purchased, `INVALID` BatchShipments must be removed<br>
 * Batches that are in the `PURCHASING` state are being purchased<br>
 * `PURCHASED` batches are finished purchasing.
 */
exports.BatchStatus = {
    Validating: "VALIDATING",
    Valid: "VALID",
    Invalid: "INVALID",
    Purchasing: "PURCHASING",
    Purchased: "PURCHASED",
};
/** @internal */
exports.ObjectResults$inboundSchema = z.pipe(z.object({
    creation_failed: z.int(),
    creation_succeeded: z.int(),
    purchase_failed: z.int(),
    purchase_succeeded: z.int(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "creation_failed": "creationFailed",
        "creation_succeeded": "creationSucceeded",
        "purchase_failed": "purchaseFailed",
        "purchase_succeeded": "purchaseSucceeded",
    });
}));
function objectResultsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ObjectResults$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObjectResults' from JSON`);
}
/** @internal */
exports.BatchStatus$inboundSchema = z
    .enum(exports.BatchStatus);
/** @internal */
exports.Batch$inboundSchema = z.pipe(z.object({
    default_carrier_account: z.string(),
    default_servicelevel_token: z.string(),
    label_filetype: z.optional(labelfiletypeenum_js_1.LabelFileTypeEnum$inboundSchema),
    metadata: z.optional(z.string()),
    batch_shipments: batchshipmentpaginatedlist_js_1.BatchShipmentPaginatedList$inboundSchema,
    label_url: z.array(z.string()),
    object_created: z.string(),
    object_id: z.string(),
    object_owner: z.string(),
    object_results: z.lazy(() => exports.ObjectResults$inboundSchema),
    object_updated: z.string(),
    status: exports.BatchStatus$inboundSchema,
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "default_carrier_account": "defaultCarrierAccount",
        "default_servicelevel_token": "defaultServicelevelToken",
        "label_filetype": "labelFiletype",
        "batch_shipments": "batchShipments",
        "label_url": "labelUrl",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_results": "objectResults",
        "object_updated": "objectUpdated",
    });
}));
function batchFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Batch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Batch' from JSON`);
}
