// shippo/models/components/billing.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Billing$outboundSchema = exports.Billing$inboundSchema = exports.Type$outboundSchema = exports.Type$inboundSchema = exports.Type = void 0;
exports.billingToJSON = billingToJSON;
exports.billingFromJSON = billingFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Party to be billed. (Leave blank for DHL Germany.)
 */
exports.Type = {
    Sender: "SENDER",
    Recipient: "RECIPIENT",
    ThirdParty: "THIRD_PARTY",
    ThirdPartyConsignee: "THIRD_PARTY_CONSIGNEE",
    Collect: "COLLECT",
};
/** @internal */
exports.Type$inboundSchema = z.enum(exports.Type);
/** @internal */
exports.Type$outboundSchema = exports.Type$inboundSchema;
/** @internal */
exports.Billing$inboundSchema = z.pipe(z.object({
    account: z.optional(z.string()),
    country: z.optional(z.string()),
    participation_code: z.optional(z.string()),
    type: z.optional(exports.Type$inboundSchema),
    zip: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "participation_code": "participationCode",
    });
}));
/** @internal */
exports.Billing$outboundSchema = z.pipe(z.object({
    account: z.optional(z.string()),
    country: z.optional(z.string()),
    participationCode: z.optional(z.string()),
    type: z.optional(exports.Type$outboundSchema),
    zip: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        participationCode: "participation_code",
    });
}));
function billingToJSON(billing) {
    return JSON.stringify(exports.Billing$outboundSchema.parse(billing));
}
function billingFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Billing$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Billing' from JSON`);
}
