// shippo/models/components/carrieraccountwithextrainfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountWithExtraInfo$inboundSchema = exports.ObjectInfo$inboundSchema = exports.Authentication$inboundSchema = exports.CarrierAccountWithExtraInfoStatus$inboundSchema = exports.CarrierAccountWithExtraInfoType$inboundSchema = exports.ParametersT$inboundSchema = exports.CarrierAccountWithExtraInfoStatus = exports.CarrierAccountWithExtraInfoType = void 0;
exports.parametersFromJSON = parametersFromJSON;
exports.authenticationFromJSON = authenticationFromJSON;
exports.objectInfoFromJSON = objectInfoFromJSON;
exports.carrierAccountWithExtraInfoFromJSON = carrierAccountWithExtraInfoFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const carrieraccountservicelevel_js_1 = require("./carrieraccountservicelevel.js");
const fedexconnectexistingownaccountparameters_js_1 = require("./fedexconnectexistingownaccountparameters.js");
const upsconnectexistingownaccountparameters_js_1 = require("./upsconnectexistingownaccountparameters.js");
/**
 * Authentication method used by this account.
 */
exports.CarrierAccountWithExtraInfoType = {
    Default: "default",
    Oauth: "oauth",
};
/**
 * Current authentication status. Possible values: 'disconnected' (authorization lost, reconnect needed), 'connected' (authorized and active), 'authorization_pending' (awaiting initial authorization flow).
 */
exports.CarrierAccountWithExtraInfoStatus = {
    Disconnected: "disconnected",
    Connected: "connected",
    AuthorizationPending: "authorization_pending",
};
/** @internal */
exports.ParametersT$inboundSchema = z
    .union([
    upsconnectexistingownaccountparameters_js_1.UPSConnectExistingOwnAccountParameters$inboundSchema,
    fedexconnectexistingownaccountparameters_js_1.FedExConnectExistingOwnAccountParameters$inboundSchema,
    z.record(z.string(), z.any()),
]);
function parametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ParametersT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ParametersT' from JSON`);
}
/** @internal */
exports.CarrierAccountWithExtraInfoType$inboundSchema = z.enum(exports.CarrierAccountWithExtraInfoType);
/** @internal */
exports.CarrierAccountWithExtraInfoStatus$inboundSchema = z.enum(exports.CarrierAccountWithExtraInfoStatus);
/** @internal */
exports.Authentication$inboundSchema = z.object({
    type: z.optional(exports.CarrierAccountWithExtraInfoType$inboundSchema),
    status: z.optional(exports.CarrierAccountWithExtraInfoStatus$inboundSchema),
});
function authenticationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Authentication$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Authentication' from JSON`);
}
/** @internal */
exports.ObjectInfo$inboundSchema = z
    .object({
    authentication: z.optional(z.lazy(() => exports.Authentication$inboundSchema)),
});
function objectInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ObjectInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ObjectInfo' from JSON`);
}
/** @internal */
exports.CarrierAccountWithExtraInfo$inboundSchema = z.pipe(z.object({
    account_id: z.string(),
    active: z.optional(z.boolean()),
    carrier: z.string(),
    parameters: z.optional(z.union([
        upsconnectexistingownaccountparameters_js_1.UPSConnectExistingOwnAccountParameters$inboundSchema,
        fedexconnectexistingownaccountparameters_js_1.FedExConnectExistingOwnAccountParameters$inboundSchema,
        z.record(z.string(), z.any()),
    ])),
    carrier_name: z.optional(z.any()),
    is_shippo_account: z.optional(z.boolean()),
    metadata: z.optional(z.string()),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    service_levels: z.optional(z.array(carrieraccountservicelevel_js_1.CarrierAccountServiceLevel$inboundSchema)),
    test: z.optional(z.boolean()),
    object_info: z.optional(z.lazy(() => exports.ObjectInfo$inboundSchema)),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "carrier_name": "carrierName",
        "is_shippo_account": "isShippoAccount",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "service_levels": "serviceLevels",
        "object_info": "objectInfo",
    });
}));
function carrierAccountWithExtraInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountWithExtraInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountWithExtraInfo' from JSON`);
}
