// shippo/models/components/carrieraccountupscreaterequestparameters.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountUPSCreateRequestParameters$outboundSchema = void 0;
exports.carrierAccountUPSCreateRequestParametersToJSON = carrierAccountUPSCreateRequestParametersToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.CarrierAccountUPSCreateRequestParameters$outboundSchema = z.pipe(z.object({
    billingAddressCity: z.string(),
    billingAddressCountryIso2: z.string(),
    billingAddressState: z.string(),
    billingAddressStreet1: z.string(),
    billingAddressStreet2: z.optional(z.string()),
    billingAddressZip: z.string(),
    company: z.optional(z.string()),
    email: z.optional(z.string()),
    fullName: z.optional(z.string()),
    phone: z.optional(z.string()),
    pickupAddressCity: z.string(),
    pickupAddressCountryIso2: z.string(),
    pickupAddressSameAsBillingAddress: z.optional(z.boolean()),
    pickupAddressState: z.string(),
    pickupAddressStreet1: z.string(),
    pickupAddressStreet2: z.optional(z.string()),
    pickupAddressZip: z.string(),
    upsAgreements: z.boolean(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        billingAddressCity: "billing_address_city",
        billingAddressCountryIso2: "billing_address_country_iso2",
        billingAddressState: "billing_address_state",
        billingAddressStreet1: "billing_address_street1",
        billingAddressStreet2: "billing_address_street2",
        billingAddressZip: "billing_address_zip",
        fullName: "full_name",
        pickupAddressCity: "pickup_address_city",
        pickupAddressCountryIso2: "pickup_address_country_iso2",
        pickupAddressSameAsBillingAddress: "pickup_address_same_as_billing_address",
        pickupAddressState: "pickup_address_state",
        pickupAddressStreet1: "pickup_address_street1",
        pickupAddressStreet2: "pickup_address_street2",
        pickupAddressZip: "pickup_address_zip",
        upsAgreements: "ups_agreements",
    });
}));
function carrierAccountUPSCreateRequestParametersToJSON(carrierAccountUPSCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountUPSCreateRequestParameters$outboundSchema.parse(carrierAccountUPSCreateRequestParameters));
}
