// shippo/models/components/cod.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cod$outboundSchema = exports.Cod$inboundSchema = exports.PaymentMethod$outboundSchema = exports.PaymentMethod$inboundSchema = exports.PaymentMethod = void 0;
exports.codToJSON = codToJSON;
exports.codFromJSON = codFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Secured funds include money orders, certified cheques and others (see
 *
 * @remarks
 * <a href="https://www.ups.com/content/us/en/shipping/time/service/value_added/cod.html">UPS</a> for details).
 * If no payment_method inputted the value defaults to "ANY".)
 */
exports.PaymentMethod = {
    SecuredFunds: "SECURED_FUNDS",
    Cash: "CASH",
    Any: "ANY",
};
/** @internal */
exports.PaymentMethod$inboundSchema = z.enum(exports.PaymentMethod);
/** @internal */
exports.PaymentMethod$outboundSchema = exports.PaymentMethod$inboundSchema;
/** @internal */
exports.Cod$inboundSchema = z.pipe(z.object({
    amount: z.optional(z.string()),
    currency: z.optional(z.string()),
    payment_method: z.optional(exports.PaymentMethod$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_method": "paymentMethod",
    });
}));
/** @internal */
exports.Cod$outboundSchema = z.pipe(z.object({
    amount: z.optional(z.string()),
    currency: z.optional(z.string()),
    paymentMethod: z.optional(exports.PaymentMethod$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentMethod: "payment_method",
    });
}));
function codToJSON(cod) {
    return JSON.stringify(exports.Cod$outboundSchema.parse(cod));
}
function codFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Cod$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Cod' from JSON`);
}
