// shippo/models/components/connectexistingownaccountrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectExistingOwnAccountRequest$outboundSchema = exports.ConnectExistingOwnAccountRequestParameters$outboundSchema = void 0;
exports.connectExistingOwnAccountRequestParametersToJSON = connectExistingOwnAccountRequestParametersToJSON;
exports.connectExistingOwnAccountRequestToJSON = connectExistingOwnAccountRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const fedexconnectexistingownaccountparameters_js_1 = require("./fedexconnectexistingownaccountparameters.js");
const upsconnectexistingownaccountparameters_js_1 = require("./upsconnectexistingownaccountparameters.js");
/** @internal */
exports.ConnectExistingOwnAccountRequestParameters$outboundSchema = z.union([
    upsconnectexistingownaccountparameters_js_1.UPSConnectExistingOwnAccountParameters$outboundSchema,
    fedexconnectexistingownaccountparameters_js_1.FedExConnectExistingOwnAccountParameters$outboundSchema,
    z.record(z.string(), z.any()),
]);
function connectExistingOwnAccountRequestParametersToJSON(connectExistingOwnAccountRequestParameters) {
    return JSON.stringify(exports.ConnectExistingOwnAccountRequestParameters$outboundSchema.parse(connectExistingOwnAccountRequestParameters));
}
/** @internal */
exports.ConnectExistingOwnAccountRequest$outboundSchema = z.pipe(z.object({
    accountId: z.string(),
    active: z.optional(z.boolean()),
    carrier: z.string(),
    metadata: z.optional(z.string()),
    parameters: z.union([
        upsconnectexistingownaccountparameters_js_1.UPSConnectExistingOwnAccountParameters$outboundSchema,
        fedexconnectexistingownaccountparameters_js_1.FedExConnectExistingOwnAccountParameters$outboundSchema,
        z.record(z.string(), z.any()),
    ]),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
    });
}));
function connectExistingOwnAccountRequestToJSON(connectExistingOwnAccountRequest) {
    return JSON.stringify(exports.ConnectExistingOwnAccountRequest$outboundSchema.parse(connectExistingOwnAccountRequest));
}
