// shippo/models/components/corerate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoreRate$outboundSchema = exports.CoreRate$inboundSchema = void 0;
exports.coreRateToJSON = coreRateToJSON;
exports.coreRateFromJSON = coreRateFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CoreRate$inboundSchema = z.pipe(z.object({
    amount: z.optional(z.string()),
    amount_local: z.optional(z.string()),
    currency: z.optional(z.string()),
    currency_local: z.optional(z.string()),
    object_id: z.optional(z.string()),
    provider: z.optional(z.string()),
    carrier_account: z.optional(z.string()),
    servicelevel_name: z.optional(z.string()),
    servicelevel_token: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_local": "amountLocal",
        "currency_local": "currencyLocal",
        "object_id": "objectId",
        "carrier_account": "carrierAccount",
        "servicelevel_name": "servicelevelName",
        "servicelevel_token": "servicelevelToken",
    });
}));
/** @internal */
exports.CoreRate$outboundSchema = z.pipe(z.object({
    amount: z.optional(z.string()),
    amountLocal: z.optional(z.string()),
    currency: z.optional(z.string()),
    currencyLocal: z.optional(z.string()),
    objectId: z.optional(z.string()),
    provider: z.optional(z.string()),
    carrierAccount: z.optional(z.string()),
    servicelevelName: z.optional(z.string()),
    servicelevelToken: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountLocal: "amount_local",
        currencyLocal: "currency_local",
        objectId: "object_id",
        carrierAccount: "carrier_account",
        servicelevelName: "servicelevel_name",
        servicelevelToken: "servicelevel_token",
    });
}));
function coreRateToJSON(coreRate) {
    return JSON.stringify(exports.CoreRate$outboundSchema.parse(coreRate));
}
function coreRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CoreRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CoreRate' from JSON`);
}
