// shippo/models/components/customsitemcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsItemCreateRequest$outboundSchema = void 0;
exports.customsItemCreateRequestToJSON = customsItemCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.CustomsItemCreateRequest$outboundSchema = z.pipe(z.object({
    description: z.string(),
    eccnEar99: z.optional(z.string()),
    massUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema,
    metadata: z.optional(z.string()),
    netWeight: z.string(),
    originCountry: z.string(),
    quantity: z.int(),
    skuCode: z.optional(z.string()),
    hsCode: z.optional(z.string()),
    tariffNumber: z.optional(z.string()),
    valueAmount: z.string(),
    valueCurrency: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        eccnEar99: "eccn_ear99",
        massUnit: "mass_unit",
        netWeight: "net_weight",
        originCountry: "origin_country",
        skuCode: "sku_code",
        hsCode: "hs_code",
        tariffNumber: "tariff_number",
        valueAmount: "value_amount",
        valueCurrency: "value_currency",
    });
}));
function customsItemCreateRequestToJSON(customsItemCreateRequest) {
    return JSON.stringify(exports.CustomsItemCreateRequest$outboundSchema.parse(customsItemCreateRequest));
}
