// shippo/models/components/customsdeclarationeelpfcenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationEelPfcEnum$outboundSchema = exports.CustomsDeclarationEelPfcEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * EEL / PFC type of the shipment. For most shipments from the US to CA, `NOEEI_30_36` is applicable; for most
 *
 * @remarks
 * other shipments from the US, `NOEEI_30_37_a` is applicable.
 * Allowed values available <a href="#tag/Customs-Declaration-EELPFC">here</a>
 */
exports.CustomsDeclarationEelPfcEnum = {
    NOEEI3037A: "NOEEI_30_37_a",
    NOEEI3037H: "NOEEI_30_37_h",
    NOEEI3037F: "NOEEI_30_37_f",
    Noeei3036: "NOEEI_30_36",
    AesItn: "AES_ITN",
};
/** @internal */
exports.CustomsDeclarationEelPfcEnum$outboundSchema = z.enum(exports.CustomsDeclarationEelPfcEnum);
