// shippo/models/components/customsdeclarationcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationCreateRequest$outboundSchema = exports.DutiesPayor$outboundSchema = exports.CustomsDeclarationCreateRequestAddress$outboundSchema = exports.CustomsDeclarationCreateRequestType$outboundSchema = exports.CustomsDeclarationCreateRequestType = void 0;
exports.customsDeclarationCreateRequestAddressToJSON = customsDeclarationCreateRequestAddressToJSON;
exports.dutiesPayorToJSON = dutiesPayorToJSON;
exports.customsDeclarationCreateRequestToJSON = customsDeclarationCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const addressimporter_js_1 = require("./addressimporter.js");
const customsdeclarationb13afilingoptionenum_js_1 = require("./customsdeclarationb13afilingoptionenum.js");
const customsdeclarationcontentstypeenum_js_1 = require("./customsdeclarationcontentstypeenum.js");
const customsdeclarationeelpfcenum_js_1 = require("./customsdeclarationeelpfcenum.js");
const customsdeclarationincotermenum_js_1 = require("./customsdeclarationincotermenum.js");
const customsdeclarationnondeliveryoptionenum_js_1 = require("./customsdeclarationnondeliveryoptionenum.js");
const customsexporteridentification_js_1 = require("./customsexporteridentification.js");
const customsitemcreaterequest_js_1 = require("./customsitemcreaterequest.js");
/**
 * Party to be billed for duties.
 */
exports.CustomsDeclarationCreateRequestType = {
    Sender: "SENDER",
    Recipient: "RECIPIENT",
    ThirdParty: "THIRD_PARTY",
};
/** @internal */
exports.CustomsDeclarationCreateRequestType$outboundSchema = z.enum(exports.CustomsDeclarationCreateRequestType);
/** @internal */
exports.CustomsDeclarationCreateRequestAddress$outboundSchema = z.object({
    name: z.optional(z.string()),
    zip: z.optional(z.string()),
    country: z.optional(z.string()),
});
function customsDeclarationCreateRequestAddressToJSON(customsDeclarationCreateRequestAddress) {
    return JSON.stringify(exports.CustomsDeclarationCreateRequestAddress$outboundSchema.parse(customsDeclarationCreateRequestAddress));
}
/** @internal */
exports.DutiesPayor$outboundSchema = z.object({
    account: z.optional(z.string()),
    type: z.optional(exports.CustomsDeclarationCreateRequestType$outboundSchema),
    address: z.optional(z.lazy(() => exports.CustomsDeclarationCreateRequestAddress$outboundSchema)),
});
function dutiesPayorToJSON(dutiesPayor) {
    return JSON.stringify(exports.DutiesPayor$outboundSchema.parse(dutiesPayor));
}
/** @internal */
exports.CustomsDeclarationCreateRequest$outboundSchema = z.pipe(z.object({
    aesItn: z.optional(z.string()),
    b13aFilingOption: z.optional(customsdeclarationb13afilingoptionenum_js_1.CustomsDeclarationB13AFilingOptionEnum$outboundSchema),
    b13aNumber: z.optional(z.string()),
    certificate: z.optional(z.string()),
    certify: z.boolean(),
    certifySigner: z.string(),
    commercialInvoice: z.optional(z.boolean()),
    contentsExplanation: z.optional(z.string()),
    disclaimer: z.optional(z.string()),
    dutiesPayor: z.optional(z.lazy(() => exports.DutiesPayor$outboundSchema)),
    exporterIdentification: z.optional(customsexporteridentification_js_1.CustomsExporterIdentification$outboundSchema),
    exporterReference: z.optional(z.string()),
    importerReference: z.optional(z.string()),
    isVatCollected: z.optional(z.boolean()),
    invoice: z.optional(z.string()),
    license: z.optional(z.string()),
    metadata: z.optional(z.string()),
    notes: z.optional(z.string()),
    addressImporter: z.optional(addressimporter_js_1.AddressImporter$outboundSchema),
    contentsType: customsdeclarationcontentstypeenum_js_1.CustomsDeclarationContentsTypeEnum$outboundSchema,
    eelPfc: z.optional(customsdeclarationeelpfcenum_js_1.CustomsDeclarationEelPfcEnum$outboundSchema),
    incoterm: z.optional(customsdeclarationincotermenum_js_1.CustomsDeclarationIncotermEnum$outboundSchema),
    items: z.array(customsitemcreaterequest_js_1.CustomsItemCreateRequest$outboundSchema),
    nonDeliveryOption: customsdeclarationnondeliveryoptionenum_js_1.CustomsDeclarationNonDeliveryOptionEnum$outboundSchema,
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        aesItn: "aes_itn",
        b13aFilingOption: "b13a_filing_option",
        b13aNumber: "b13a_number",
        certifySigner: "certify_signer",
        commercialInvoice: "commercial_invoice",
        contentsExplanation: "contents_explanation",
        dutiesPayor: "duties_payor",
        exporterIdentification: "exporter_identification",
        exporterReference: "exporter_reference",
        importerReference: "importer_reference",
        isVatCollected: "is_vat_collected",
        addressImporter: "address_importer",
        contentsType: "contents_type",
        eelPfc: "eel_pfc",
        nonDeliveryOption: "non_delivery_option",
    });
}));
function customsDeclarationCreateRequestToJSON(customsDeclarationCreateRequest) {
    return JSON.stringify(exports.CustomsDeclarationCreateRequest$outboundSchema.parse(customsDeclarationCreateRequest));
}
