// shippo/models/components/customsitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsItem$inboundSchema = void 0;
exports.customsItemFromJSON = customsItemFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.CustomsItem$inboundSchema = z
    .pipe(z.object({
    description: z.string(),
    eccn_ear99: z.optional(z.string()),
    mass_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema,
    metadata: z.optional(z.string()),
    net_weight: z.string(),
    origin_country: z.string(),
    quantity: z.int(),
    sku_code: z.optional(z.string()),
    hs_code: z.optional(z.string()),
    tariff_number: z.optional(z.string()),
    value_amount: z.string(),
    value_currency: z.string(),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    object_state: z.optional(objectstateenum_js_1.ObjectStateEnum$inboundSchema),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "eccn_ear99": "eccnEar99",
        "mass_unit": "massUnit",
        "net_weight": "netWeight",
        "origin_country": "originCountry",
        "sku_code": "skuCode",
        "hs_code": "hsCode",
        "tariff_number": "tariffNumber",
        "value_amount": "valueAmount",
        "value_currency": "valueCurrency",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_state": "objectState",
        "object_updated": "objectUpdated",
    });
}));
function customsItemFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsItem' from JSON`);
}
