// shippo/models/components/customsdeclaration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclaration$inboundSchema = exports.CustomsDeclarationDutiesPayor$inboundSchema = exports.CustomsDeclarationAddress$inboundSchema = exports.CustomsDeclarationType$inboundSchema = exports.CustomsDeclarationType = void 0;
exports.customsDeclarationAddressFromJSON = customsDeclarationAddressFromJSON;
exports.customsDeclarationDutiesPayorFromJSON = customsDeclarationDutiesPayorFromJSON;
exports.customsDeclarationFromJSON = customsDeclarationFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customsexporteridentification_js_1 = require("./customsexporteridentification.js");
const customsinvoicedcharges_js_1 = require("./customsinvoicedcharges.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
/**
 * Party to be billed for duties.
 */
exports.CustomsDeclarationType = {
    Sender: "SENDER",
    Recipient: "RECIPIENT",
    ThirdParty: "THIRD_PARTY",
};
/** @internal */
exports.CustomsDeclarationType$inboundSchema = z.enum(exports.CustomsDeclarationType);
/** @internal */
exports.CustomsDeclarationAddress$inboundSchema = z.object({
    name: z.optional(z.string()),
    zip: z.optional(z.string()),
    country: z.optional(z.string()),
});
function customsDeclarationAddressFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsDeclarationAddress$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsDeclarationAddress' from JSON`);
}
/** @internal */
exports.CustomsDeclarationDutiesPayor$inboundSchema = z.object({
    account: z.optional(z.string()),
    type: z.optional(exports.CustomsDeclarationType$inboundSchema),
    address: z.optional(z.lazy(() => exports.CustomsDeclarationAddress$inboundSchema)),
});
function customsDeclarationDutiesPayorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsDeclarationDutiesPayor$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsDeclarationDutiesPayor' from JSON`);
}
/** @internal */
exports.CustomsDeclaration$inboundSchema = z.pipe(z.object({
    aes_itn: z.optional(z.string()),
    b13a_filing_option: z.optional(z.string()),
    b13a_number: z.optional(z.string()),
    certificate: z.optional(z.string()),
    certify: z.boolean(),
    certify_signer: z.string(),
    commercial_invoice: z.optional(z.boolean()),
    contents_explanation: z.optional(z.string()),
    disclaimer: z.optional(z.string()),
    duties_payor: z.optional(z.lazy(() => exports.CustomsDeclarationDutiesPayor$inboundSchema)),
    exporter_identification: z.optional(customsexporteridentification_js_1.CustomsExporterIdentification$inboundSchema),
    exporter_reference: z.optional(z.string()),
    importer_reference: z.optional(z.string()),
    is_vat_collected: z.optional(z.boolean()),
    invoice: z.optional(z.string()),
    license: z.optional(z.string()),
    metadata: z.optional(z.string()),
    notes: z.optional(z.string()),
    address_importer: z.optional(z.string()),
    contents_type: z.string(),
    eel_pfc: z.optional(z.string()),
    incoterm: z.optional(z.string()),
    invoiced_charges: z.optional(customsinvoicedcharges_js_1.CustomsInvoicedCharges$inboundSchema),
    items: z.array(z.string()),
    non_delivery_option: z.string(),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    object_state: z.optional(objectstateenum_js_1.ObjectStateEnum$inboundSchema),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "aes_itn": "aesItn",
        "b13a_filing_option": "b13aFilingOption",
        "b13a_number": "b13aNumber",
        "certify_signer": "certifySigner",
        "commercial_invoice": "commercialInvoice",
        "contents_explanation": "contentsExplanation",
        "duties_payor": "dutiesPayor",
        "exporter_identification": "exporterIdentification",
        "exporter_reference": "exporterReference",
        "importer_reference": "importerReference",
        "is_vat_collected": "isVatCollected",
        "address_importer": "addressImporter",
        "contents_type": "contentsType",
        "eel_pfc": "eelPfc",
        "invoiced_charges": "invoicedCharges",
        "non_delivery_option": "nonDeliveryOption",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_state": "objectState",
        "object_updated": "objectUpdated",
    });
}));
function customsDeclarationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsDeclaration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsDeclaration' from JSON`);
}
