// shippo/models/components/customstaxidentification.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsTaxIdentification$outboundSchema = exports.CustomsTaxIdentification$inboundSchema = exports.CustomsTaxIdentificationType$outboundSchema = exports.CustomsTaxIdentificationType$inboundSchema = exports.CustomsTaxIdentificationType = void 0;
exports.customsTaxIdentificationToJSON = customsTaxIdentificationToJSON;
exports.customsTaxIdentificationFromJSON = customsTaxIdentificationFromJSON;
const z = __importStar(require("zod/v4-mini"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Type of tax identification.
 *
 * @remarks
 * * `EIN` - Employer Identification Number, also known as a Federal Tax Identification Number.
 * * `VAT` - Value Added Tax identification number.
 * * `IOSS` - Import One-Stop Shop
 * * `ARN` - Australian Taxation Office Reference Number
 */
exports.CustomsTaxIdentificationType = {
    Ein: "EIN",
    Vat: "VAT",
    Ioss: "IOSS",
    Arn: "ARN",
};
/** @internal */
exports.CustomsTaxIdentificationType$inboundSchema = z.enum(exports.CustomsTaxIdentificationType);
/** @internal */
exports.CustomsTaxIdentificationType$outboundSchema = exports.CustomsTaxIdentificationType$inboundSchema;
/** @internal */
exports.CustomsTaxIdentification$inboundSchema = z.object({
    number: z.optional(z.string()),
    type: z.optional(exports.CustomsTaxIdentificationType$inboundSchema),
});
/** @internal */
exports.CustomsTaxIdentification$outboundSchema = z.object({
    number: z.optional(z.string()),
    type: z.optional(exports.CustomsTaxIdentificationType$outboundSchema),
});
function customsTaxIdentificationToJSON(customsTaxIdentification) {
    return JSON.stringify(exports.CustomsTaxIdentification$outboundSchema.parse(customsTaxIdentification));
}
function customsTaxIdentificationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsTaxIdentification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsTaxIdentification' from JSON`);
}
