// shippo/models/components/dryice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DryIce$outboundSchema = exports.DryIce$inboundSchema = void 0;
exports.dryIceToJSON = dryIceToJSON;
exports.dryIceFromJSON = dryIceFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DryIce$inboundSchema = z.pipe(z.object({
    contains_dry_ice: z.optional(z.boolean()),
    weight: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "contains_dry_ice": "containsDryIce",
    });
}));
/** @internal */
exports.DryIce$outboundSchema = z
    .pipe(z.object({
    containsDryIce: z.optional(z.boolean()),
    weight: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        containsDryIce: "contains_dry_ice",
    });
}));
function dryIceToJSON(dryIce) {
    return JSON.stringify(exports.DryIce$outboundSchema.parse(dryIce));
}
function dryIceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DryIce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DryIce' from JSON`);
}
