// shippo/models/components/instanttransactioncreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstantTransactionCreateRequest$outboundSchema = exports.LabelFileType$outboundSchema = exports.LabelFileType = void 0;
exports.instantTransactionCreateRequestToJSON = instantTransactionCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const shipmentcreaterequest_js_1 = require("./shipmentcreaterequest.js");
exports.LabelFileType = {
    Png: "PNG",
    PNG23x75: "PNG_2.3x7.5",
    Pdf: "PDF",
    PDF23x75: "PDF_2.3x7.5",
    PDF4x6: "PDF_4x6",
    PDF4x8: "PDF_4x8",
    PdfA4: "PDF_A4",
    PdfA6: "PDF_A6",
    Zplii: "ZPLII",
};
/** @internal */
exports.LabelFileType$outboundSchema = z.enum(exports.LabelFileType);
/** @internal */
exports.InstantTransactionCreateRequest$outboundSchema = z.pipe(z.object({
    async: z._default(z.boolean(), false),
    carrierAccount: z.string(),
    labelFileType: z.optional(exports.LabelFileType$outboundSchema),
    metadata: z.optional(z.string()),
    servicelevelToken: z.string(),
    shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$outboundSchema,
    order: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        labelFileType: "label_file_type",
        servicelevelToken: "servicelevel_token",
    });
}));
function instantTransactionCreateRequestToJSON(instantTransactionCreateRequest) {
    return JSON.stringify(exports.InstantTransactionCreateRequest$outboundSchema.parse(instantTransactionCreateRequest));
}
