// shippo/models/components/labelfiletypeenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelFileTypeEnum$outboundSchema = exports.LabelFileTypeEnum$inboundSchema = exports.LabelFileTypeEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * Print format of the <a href="https://docs.goshippo.com/docs/shipments/shippinglabelsizes/">label</a>. If empty, will use the default format set from
 *
 * @remarks
 * <a href="https://apps.goshippo.com/settings/labels">the Shippo dashboard.</a>
 */
exports.LabelFileTypeEnum = {
    Png: "PNG",
    PNG23x75: "PNG_2.3x7.5",
    Pdf: "PDF",
    PDF23x75: "PDF_2.3x7.5",
    PDF4x6: "PDF_4x6",
    PDF4x8: "PDF_4x8",
    PdfA4: "PDF_A4",
    PdfA5: "PDF_A5",
    PdfA6: "PDF_A6",
    Zplii: "ZPLII",
};
/** @internal */
exports.LabelFileTypeEnum$inboundSchema = z.enum(exports.LabelFileTypeEnum);
/** @internal */
exports.LabelFileTypeEnum$outboundSchema = exports.LabelFileTypeEnum$inboundSchema;
