// shippo/models/components/lineitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineItem$outboundSchema = exports.LineItem$inboundSchema = void 0;
exports.lineItemToJSON = lineItemToJSON;
exports.lineItemFromJSON = lineItemFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.LineItem$inboundSchema = z.pipe(z.object({
    currency: z.optional(z.string()),
    manufacture_country: z.optional(z.string()),
    max_delivery_time: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    max_ship_time: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    quantity: z.optional(z.int()),
    sku: z.optional(z.string()),
    title: z.optional(z.string()),
    total_price: z.optional(z.string()),
    variant_title: z.optional(z.string()),
    weight: z.optional(z.string()),
    weight_unit: z.optional(weightunitenum_js_1.WeightUnitEnum$inboundSchema),
    object_id: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "manufacture_country": "manufactureCountry",
        "max_delivery_time": "maxDeliveryTime",
        "max_ship_time": "maxShipTime",
        "total_price": "totalPrice",
        "variant_title": "variantTitle",
        "weight_unit": "weightUnit",
        "object_id": "objectId",
    });
}));
/** @internal */
exports.LineItem$outboundSchema = z.pipe(z.object({
    currency: z.optional(z.string()),
    manufactureCountry: z.optional(z.string()),
    maxDeliveryTime: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    maxShipTime: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    quantity: z.optional(z.int()),
    sku: z.optional(z.string()),
    title: z.optional(z.string()),
    totalPrice: z.optional(z.string()),
    variantTitle: z.optional(z.string()),
    weight: z.optional(z.string()),
    weightUnit: z.optional(weightunitenum_js_1.WeightUnitEnum$outboundSchema),
    objectId: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        manufactureCountry: "manufacture_country",
        maxDeliveryTime: "max_delivery_time",
        maxShipTime: "max_ship_time",
        totalPrice: "total_price",
        variantTitle: "variant_title",
        weightUnit: "weight_unit",
        objectId: "object_id",
    });
}));
function lineItemToJSON(lineItem) {
    return JSON.stringify(exports.LineItem$outboundSchema.parse(lineItem));
}
function lineItemFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LineItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LineItem' from JSON`);
}
