// shippo/models/components/liveratecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveRateCreateRequest$outboundSchema = exports.LiveRateCreateRequestParcel$outboundSchema = exports.LiveRateCreateRequestAddressTo$outboundSchema = exports.LiveRateCreateRequestAddressFrom$outboundSchema = void 0;
exports.liveRateCreateRequestAddressFromToJSON = liveRateCreateRequestAddressFromToJSON;
exports.liveRateCreateRequestAddressToToJSON = liveRateCreateRequestAddressToToJSON;
exports.liveRateCreateRequestParcelToJSON = liveRateCreateRequestParcelToJSON;
exports.liveRateCreateRequestToJSON = liveRateCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const addresscompletecreaterequest_js_1 = require("./addresscompletecreaterequest.js");
const lineitem_js_1 = require("./lineitem.js");
const parcel_js_1 = require("./parcel.js");
/** @internal */
exports.LiveRateCreateRequestAddressFrom$outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()]);
function liveRateCreateRequestAddressFromToJSON(liveRateCreateRequestAddressFrom) {
    return JSON.stringify(exports.LiveRateCreateRequestAddressFrom$outboundSchema.parse(liveRateCreateRequestAddressFrom));
}
/** @internal */
exports.LiveRateCreateRequestAddressTo$outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()]);
function liveRateCreateRequestAddressToToJSON(liveRateCreateRequestAddressTo) {
    return JSON.stringify(exports.LiveRateCreateRequestAddressTo$outboundSchema.parse(liveRateCreateRequestAddressTo));
}
/** @internal */
exports.LiveRateCreateRequestParcel$outboundSchema = z.union([parcel_js_1.Parcel$outboundSchema, z.string()]);
function liveRateCreateRequestParcelToJSON(liveRateCreateRequestParcel) {
    return JSON.stringify(exports.LiveRateCreateRequestParcel$outboundSchema.parse(liveRateCreateRequestParcel));
}
/** @internal */
exports.LiveRateCreateRequest$outboundSchema = z.pipe(z.object({
    addressFrom: z.optional(z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()])),
    addressTo: z.union([
        addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema,
        z.string(),
    ]),
    lineItems: z.array(lineitem_js_1.LineItem$outboundSchema),
    parcel: z.optional(z.union([parcel_js_1.Parcel$outboundSchema, z.string()])),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressFrom: "address_from",
        addressTo: "address_to",
        lineItems: "line_items",
    });
}));
function liveRateCreateRequestToJSON(liveRateCreateRequest) {
    return JSON.stringify(exports.LiveRateCreateRequest$outboundSchema.parse(liveRateCreateRequest));
}
