// shippo/models/components/manifestcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManifestCreateRequest$outboundSchema = exports.ManifestCreateRequestAddressFrom$outboundSchema = void 0;
exports.manifestCreateRequestAddressFromToJSON = manifestCreateRequestAddressFromToJSON;
exports.manifestCreateRequestToJSON = manifestCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
/** @internal */
exports.ManifestCreateRequestAddressFrom$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
function manifestCreateRequestAddressFromToJSON(manifestCreateRequestAddressFrom) {
    return JSON.stringify(exports.ManifestCreateRequestAddressFrom$outboundSchema.parse(manifestCreateRequestAddressFrom));
}
/** @internal */
exports.ManifestCreateRequest$outboundSchema = z.pipe(z.object({
    carrierAccount: z.string(),
    shipmentDate: z.string(),
    transactions: z.optional(z.array(z.string())),
    addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    async: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        shipmentDate: "shipment_date",
        addressFrom: "address_from",
    });
}));
function manifestCreateRequestToJSON(manifestCreateRequest) {
    return JSON.stringify(exports.ManifestCreateRequest$outboundSchema.parse(manifestCreateRequest));
}
