// shippo/models/components/manifest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Manifest$inboundSchema = exports.ManifestStatus$inboundSchema = exports.ManifestStatus = void 0;
exports.manifestFromJSON = manifestFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Indicates the status of the manifest.
 */
exports.ManifestStatus = {
    Queued: "QUEUED",
    Success: "SUCCESS",
    Error: "ERROR",
};
/** @internal */
exports.ManifestStatus$inboundSchema = z.enum(exports.ManifestStatus);
/** @internal */
exports.Manifest$inboundSchema = z.pipe(z.object({
    carrier_account: z.string(),
    shipment_date: z.string(),
    transactions: z.optional(z.array(z.string())),
    address_from: z.string(),
    documents: z.array(z.string()),
    errors: z.optional(z.array(z.string())),
    object_created: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    object_id: z.string(),
    object_owner: z.string(),
    object_updated: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    status: exports.ManifestStatus$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "shipment_date": "shipmentDate",
        "address_from": "addressFrom",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_updated": "objectUpdated",
    });
}));
function manifestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Manifest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Manifest' from JSON`);
}
