// shippo/models/components/order.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order$inboundSchema = exports.Transactions$inboundSchema = void 0;
exports.transactionsFromJSON = transactionsFromJSON;
exports.orderFromJSON = orderFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const lineitem_js_1 = require("./lineitem.js");
const ordershopappenum_js_1 = require("./ordershopappenum.js");
const orderstatusenum_js_1 = require("./orderstatusenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.Transactions$inboundSchema = z.object({});
function transactionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Transactions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Transactions' from JSON`);
}
/** @internal */
exports.Order$inboundSchema = z.pipe(z.object({
    currency: z.optional(z.string()),
    notes: z.optional(z.string()),
    order_number: z.optional(z.string()),
    order_status: z.optional(orderstatusenum_js_1.OrderStatusEnum$inboundSchema),
    placed_at: z.string(),
    shipping_cost: z.optional(z.string()),
    shipping_cost_currency: z.optional(z.string()),
    shipping_method: z.optional(z.string()),
    subtotal_price: z.optional(z.string()),
    total_price: z.optional(z.string()),
    total_tax: z.optional(z.string()),
    weight: z.optional(z.string()),
    weight_unit: z.optional(weightunitenum_js_1.WeightUnitEnum$inboundSchema),
    from_address: z.optional(address_js_1.Address$inboundSchema),
    to_address: address_js_1.Address$inboundSchema,
    line_items: z.optional(z.array(lineitem_js_1.LineItem$inboundSchema)),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    shop_app: z.optional(ordershopappenum_js_1.OrderShopAppEnum$inboundSchema),
    transactions: z.optional(z.array(z.lazy(() => exports.Transactions$inboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "order_number": "orderNumber",
        "order_status": "orderStatus",
        "placed_at": "placedAt",
        "shipping_cost": "shippingCost",
        "shipping_cost_currency": "shippingCostCurrency",
        "shipping_method": "shippingMethod",
        "subtotal_price": "subtotalPrice",
        "total_price": "totalPrice",
        "total_tax": "totalTax",
        "weight_unit": "weightUnit",
        "from_address": "fromAddress",
        "to_address": "toAddress",
        "line_items": "lineItems",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "shop_app": "shopApp",
    });
}));
function orderFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Order$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Order' from JSON`);
}
