// shippo/models/components/orderstatusenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderStatusEnum$outboundSchema = exports.OrderStatusEnum$inboundSchema = exports.OrderStatusEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * Current state of the order. See the <a href="https://docs.goshippo.com/docs/orders/orders/">orders tutorial</a>
 *
 * @remarks
 * for the logic of how the status is handled.
 */
exports.OrderStatusEnum = {
    Unknown: "UNKNOWN",
    Awaitpay: "AWAITPAY",
    Paid: "PAID",
    Refunded: "REFUNDED",
    Cancelled: "CANCELLED",
    PartiallyFulfilled: "PARTIALLY_FULFILLED",
    Shipped: "SHIPPED",
};
/** @internal */
exports.OrderStatusEnum$inboundSchema = z.enum(exports.OrderStatusEnum);
/** @internal */
exports.OrderStatusEnum$outboundSchema = exports.OrderStatusEnum$inboundSchema;
