// shippo/models/components/ordercreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderCreateRequest$outboundSchema = void 0;
exports.orderCreateRequestToJSON = orderCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const lineitembase_js_1 = require("./lineitembase.js");
const orderstatusenum_js_1 = require("./orderstatusenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.OrderCreateRequest$outboundSchema = z.pipe(z.object({
    currency: z.optional(z.string()),
    notes: z.optional(z.string()),
    orderNumber: z.optional(z.string()),
    orderStatus: z.optional(orderstatusenum_js_1.OrderStatusEnum$outboundSchema),
    placedAt: z.string(),
    shippingCost: z.optional(z.string()),
    shippingCostCurrency: z.optional(z.string()),
    shippingMethod: z.optional(z.string()),
    subtotalPrice: z.optional(z.string()),
    totalPrice: z.optional(z.string()),
    totalTax: z.optional(z.string()),
    weight: z.optional(z.string()),
    weightUnit: z.optional(weightunitenum_js_1.WeightUnitEnum$outboundSchema),
    fromAddress: z.optional(addresscreaterequest_js_1.AddressCreateRequest$outboundSchema),
    toAddress: addresscreaterequest_js_1.AddressCreateRequest$outboundSchema,
    lineItems: z.optional(z.array(lineitembase_js_1.LineItemBase$outboundSchema)),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        orderNumber: "order_number",
        orderStatus: "order_status",
        placedAt: "placed_at",
        shippingCost: "shipping_cost",
        shippingCostCurrency: "shipping_cost_currency",
        shippingMethod: "shipping_method",
        subtotalPrice: "subtotal_price",
        totalPrice: "total_price",
        totalTax: "total_tax",
        weightUnit: "weight_unit",
        fromAddress: "from_address",
        toAddress: "to_address",
        lineItems: "line_items",
    });
}));
function orderCreateRequestToJSON(orderCreateRequest) {
    return JSON.stringify(exports.OrderCreateRequest$outboundSchema.parse(orderCreateRequest));
}
