// shippo/models/components/parcelextra.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelExtra$outboundSchema = exports.ParcelExtra$inboundSchema = void 0;
exports.parcelExtraToJSON = parcelExtraToJSON;
exports.parcelExtraFromJSON = parcelExtraFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const cod_js_1 = require("./cod.js");
const parcelinsurance_js_1 = require("./parcelinsurance.js");
/** @internal */
exports.ParcelExtra$inboundSchema = z
    .pipe(z.object({
    COD: z.optional(cod_js_1.Cod$inboundSchema),
    insurance: z.optional(parcelinsurance_js_1.ParcelInsurance$inboundSchema),
    reference_1: z.optional(z.string()),
    reference_2: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "COD": "cod",
        "reference_1": "reference1",
        "reference_2": "reference2",
    });
}));
/** @internal */
exports.ParcelExtra$outboundSchema = z.pipe(z.object({
    cod: z.optional(cod_js_1.Cod$outboundSchema),
    insurance: z.optional(parcelinsurance_js_1.ParcelInsurance$outboundSchema),
    reference1: z.optional(z.string()),
    reference2: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        cod: "COD",
        reference1: "reference_1",
        reference2: "reference_2",
    });
}));
function parcelExtraToJSON(parcelExtra) {
    return JSON.stringify(exports.ParcelExtra$outboundSchema.parse(parcelExtra));
}
function parcelExtraFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ParcelExtra$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ParcelExtra' from JSON`);
}
