// shippo/models/components/parcelinsurance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelInsurance$outboundSchema = exports.ParcelInsurance$inboundSchema = exports.ParcelInsuranceProvider$outboundSchema = exports.ParcelInsuranceProvider$inboundSchema = exports.ParcelInsuranceProvider = void 0;
exports.parcelInsuranceToJSON = parcelInsuranceToJSON;
exports.parcelInsuranceFromJSON = parcelInsuranceFromJSON;
const z = __importStar(require("zod/v4-mini"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * To have insurance cover provided by a carrier directly instead of Shippo's provider (XCover), set provider to `FEDEX`, `UPS`, or `ONTRAC`.
 */
exports.ParcelInsuranceProvider = {
    Fedex: "FEDEX",
    Ups: "UPS",
    Ontrac: "ONTRAC",
};
/** @internal */
exports.ParcelInsuranceProvider$inboundSchema = z.enum(exports.ParcelInsuranceProvider);
/** @internal */
exports.ParcelInsuranceProvider$outboundSchema = exports.ParcelInsuranceProvider$inboundSchema;
/** @internal */
exports.ParcelInsurance$inboundSchema = z.object({
    amount: z.optional(z.string()),
    content: z.optional(z.string()),
    currency: z.optional(z.string()),
    provider: z.optional(exports.ParcelInsuranceProvider$inboundSchema),
});
/** @internal */
exports.ParcelInsurance$outboundSchema = z.object({
    amount: z.optional(z.string()),
    content: z.optional(z.string()),
    currency: z.optional(z.string()),
    provider: z.optional(exports.ParcelInsuranceProvider$outboundSchema),
});
function parcelInsuranceToJSON(parcelInsurance) {
    return JSON.stringify(exports.ParcelInsurance$outboundSchema.parse(parcelInsurance));
}
function parcelInsuranceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ParcelInsurance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ParcelInsurance' from JSON`);
}
