// shippo/models/components/parceltemplatearamexaustraliaenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelTemplateAramexAustraliaEnum$outboundSchema = exports.ParcelTemplateAramexAustraliaEnum$inboundSchema = exports.ParcelTemplateAramexAustraliaEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Name | Dimensions|
 *
 * @remarks
 * |:---|:---|:---|
 * | Fastway_Australia_Satchel_A2 | Satchel A2 | 594.00 x 420.00 x 48.00 mm|
 * | Fastway_Australia_Satchel_A3 | Satchel A3 | 420.00 x 297.00 x 64.00 mm|
 * | Fastway_Australia_Satchel_A4 | Satchel A4 | 297.00 x 210.00 x 64.00 mm|
 * | Fastway_Australia_Satchel_A5 | Satchel A5 | 210.00 x 148.00 x 64.00 mm|
 */
exports.ParcelTemplateAramexAustraliaEnum = {
    FastwayAustraliaSatchelA2: "Fastway_Australia_Satchel_A2",
    FastwayAustraliaSatchelA3: "Fastway_Australia_Satchel_A3",
    FastwayAustraliaSatchelA4: "Fastway_Australia_Satchel_A4",
    FastwayAustraliaSatchelA5: "Fastway_Australia_Satchel_A5",
};
/** @internal */
exports.ParcelTemplateAramexAustraliaEnum$inboundSchema = z.enum(exports.ParcelTemplateAramexAustraliaEnum);
/** @internal */
exports.ParcelTemplateAramexAustraliaEnum$outboundSchema = exports.ParcelTemplateAramexAustraliaEnum$inboundSchema;
