// shippo/models/components/parceltemplatefedexenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelTemplateFedExEnum$outboundSchema = exports.ParcelTemplateFedExEnum$inboundSchema = exports.ParcelTemplateFedExEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Name | Dimensions|
 *
 * @remarks
 * |:---|:---|:---|
 * | FedEx_Box_10kg | FedEx® 10kg Box | 15.81 x 12.94 x 10.19 in|
 * | FedEx_Box_25kg | FedEx® 25kg Box | 54.80 x 42.10 x 33.50 in|
 * | FedEx_Box_Extra_Large_1 | FedEx® Extra Large Box (X1) | 11.88 x 11.00 x 10.75 in|
 * | FedEx_Box_Extra_Large_2 | FedEx® Extra Large Box (X2) | 15.75 x 14.13 x 6.00 in|
 * | FedEx_Box_Large_1 | FedEx® Large Box (L1) | 17.50 x 12.38 x 3.00 in|
 * | FedEx_Box_Large_2 | FedEx® Large Box (L2) | 11.25 x 8.75 x 7.75 in|
 * | FedEx_Box_Medium_1 | FedEx® Medium Box (M1) | 13.25 x 11.50 x 2.38 in|
 * | FedEx_Box_Medium_2 | FedEx® Medium Box (M2) | 11.25 x 8.75 x 4.38 in|
 * | FedEx_Box_Small_1 | FedEx® Small Box (S1) | 12.38 x 10.88 x 1.50 in|
 * | FedEx_Box_Small_2 | FedEx® Small Box (S2) | 11.25 x 8.75 x 4.38 in|
 * | FedEx_Envelope | FedEx® Envelope | 12.50 x 9.50 x 0.80 in|
 * | FedEx_Padded_Pak | FedEx® Padded Pak | 11.75 x 14.75 x 2.00 in|
 * | FedEx_Pak_1 | FedEx® Large Pak | 15.50 x 12.00 x 0.80 in|
 * | FedEx_Pak_2 | FedEx® Small Pak | 12.75 x 10.25 x 0.80 in|
 * | FedEx_Tube | FedEx® Tube | 38.00 x 6.00 x 6.00 in|
 * | FedEx_XL_Pak | FedEx® Extra Large Pak | 17.50 x 20.75 x 2.00 in|
 */
exports.ParcelTemplateFedExEnum = {
    FedExBox10kg: "FedEx_Box_10kg",
    FedExBox25kg: "FedEx_Box_25kg",
    FedExBoxExtraLarge1: "FedEx_Box_Extra_Large_1",
    FedExBoxExtraLarge2: "FedEx_Box_Extra_Large_2",
    FedExBoxLarge1: "FedEx_Box_Large_1",
    FedExBoxLarge2: "FedEx_Box_Large_2",
    FedExBoxMedium1: "FedEx_Box_Medium_1",
    FedExBoxMedium2: "FedEx_Box_Medium_2",
    FedExBoxSmall1: "FedEx_Box_Small_1",
    FedExBoxSmall2: "FedEx_Box_Small_2",
    FedExEnvelope: "FedEx_Envelope",
    FedExPaddedPak: "FedEx_Padded_Pak",
    FedExPak1: "FedEx_Pak_1",
    FedExPak2: "FedEx_Pak_2",
    FedExTube: "FedEx_Tube",
    FedExXLPak: "FedEx_XL_Pak",
};
/** @internal */
exports.ParcelTemplateFedExEnum$inboundSchema = z.enum(exports.ParcelTemplateFedExEnum);
/** @internal */
exports.ParcelTemplateFedExEnum$outboundSchema = exports.ParcelTemplateFedExEnum$inboundSchema;
