// shippo/models/components/parcel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Parcel$outboundSchema = exports.Parcel$inboundSchema = exports.ObjectState$outboundSchema = exports.ObjectState$inboundSchema = exports.ObjectState = void 0;
exports.parcelToJSON = parcelToJSON;
exports.parcelFromJSON = parcelFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
const parcelextra_js_1 = require("./parcelextra.js");
const parceltemplateenumset_js_1 = require("./parceltemplateenumset.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/**
 * A Parcel will only be valid when all required values have been sent and validated successfully.
 */
exports.ObjectState = {
    Valid: "VALID",
};
/** @internal */
exports.ObjectState$inboundSchema = z
    .enum(exports.ObjectState);
/** @internal */
exports.ObjectState$outboundSchema = exports.ObjectState$inboundSchema;
/** @internal */
exports.Parcel$inboundSchema = z.pipe(z.object({
    extra: z.optional(parcelextra_js_1.ParcelExtra$inboundSchema),
    metadata: z.optional(z.string()),
    mass_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema,
    weight: z.string(),
    distance_unit: distanceunitenum_js_1.DistanceUnitEnum$inboundSchema,
    height: z.string(),
    length: z.string(),
    width: z.string(),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    object_state: z.optional(exports.ObjectState$inboundSchema),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    template: z.optional(parceltemplateenumset_js_1.ParcelTemplateEnumSet$inboundSchema),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "mass_unit": "massUnit",
        "distance_unit": "distanceUnit",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_state": "objectState",
        "object_updated": "objectUpdated",
    });
}));
/** @internal */
exports.Parcel$outboundSchema = z
    .pipe(z.object({
    extra: z.optional(parcelextra_js_1.ParcelExtra$outboundSchema),
    metadata: z.optional(z.string()),
    massUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema,
    weight: z.string(),
    distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$outboundSchema,
    height: z.string(),
    length: z.string(),
    width: z.string(),
    objectCreated: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    objectId: z.optional(z.string()),
    objectOwner: z.optional(z.string()),
    objectState: z.optional(exports.ObjectState$outboundSchema),
    objectUpdated: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    template: z.optional(parceltemplateenumset_js_1.ParcelTemplateEnumSet$outboundSchema),
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        massUnit: "mass_unit",
        distanceUnit: "distance_unit",
        objectCreated: "object_created",
        objectId: "object_id",
        objectOwner: "object_owner",
        objectState: "object_state",
        objectUpdated: "object_updated",
    });
}));
function parcelToJSON(parcel) {
    return JSON.stringify(exports.Parcel$outboundSchema.parse(parcel));
}
function parcelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Parcel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Parcel' from JSON`);
}
