// shippo/models/components/pickup.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pickup$inboundSchema = exports.PickupStatus$inboundSchema = exports.PickupStatus = void 0;
exports.pickupFromJSON = pickupFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const location_js_1 = require("./location.js");
/**
 * Indicates the status of the pickup.
 */
exports.PickupStatus = {
    Pending: "PENDING",
    Confirmed: "CONFIRMED",
    Error: "ERROR",
    Cancelled: "CANCELLED",
};
/** @internal */
exports.PickupStatus$inboundSchema = z
    .enum(exports.PickupStatus);
/** @internal */
exports.Pickup$inboundSchema = z.pipe(z.object({
    carrier_account: z.string(),
    location: location_js_1.Location$inboundSchema,
    metadata: z.optional(z.string()),
    requested_end_time: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    requested_start_time: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    transactions: z.array(z.string()),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    confirmed_start_time: z.optional(z.string()),
    confirmed_end_time: z.optional(z.string()),
    cancel_by_time: z.optional(z.string()),
    status: z.optional(exports.PickupStatus$inboundSchema),
    confirmation_code: z.optional(z.string()),
    timezone: z.optional(z.string()),
    messages: z.optional(z.array(z.string())),
    is_test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "requested_end_time": "requestedEndTime",
        "requested_start_time": "requestedStartTime",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_updated": "objectUpdated",
        "confirmed_start_time": "confirmedStartTime",
        "confirmed_end_time": "confirmedEndTime",
        "cancel_by_time": "cancelByTime",
        "confirmation_code": "confirmationCode",
        "is_test": "isTest",
    });
}));
function pickupFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Pickup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Pickup' from JSON`);
}
