// shippo/models/components/rate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rate$inboundSchema = exports.Attributes$inboundSchema = exports.Attributes = void 0;
exports.rateFromJSON = rateFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const responsemessage_js_1 = require("./responsemessage.js");
const servicelevelwithparent_js_1 = require("./servicelevelwithparent.js");
exports.Attributes = {
    Bestvalue: "BESTVALUE",
    Cheapest: "CHEAPEST",
    Fastest: "FASTEST",
};
/** @internal */
exports.Attributes$inboundSchema = z
    .enum(exports.Attributes);
/** @internal */
exports.Rate$inboundSchema = z.pipe(z.object({
    amount: z.string(),
    amount_local: z.string(),
    currency: z.string(),
    currency_local: z.string(),
    arrives_by: z.optional(z.string()),
    attributes: z.array(exports.Attributes$inboundSchema),
    carrier_account: z.string(),
    duration_terms: z.optional(z.string()),
    estimated_days: z.optional(z.int()),
    included_insurance_price: z.optional(z.string()),
    messages: z.optional(z.array(responsemessage_js_1.ResponseMessage$inboundSchema)),
    object_created: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    object_id: z.string(),
    object_owner: z.string(),
    provider: z.string(),
    provider_image_75: z.optional(z.string()),
    provider_image_200: z.optional(z.string()),
    servicelevel: servicelevelwithparent_js_1.ServiceLevelWithParent$inboundSchema,
    shipment: z.string(),
    test: z.optional(z.boolean()),
    zone: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_local": "amountLocal",
        "currency_local": "currencyLocal",
        "arrives_by": "arrivesBy",
        "carrier_account": "carrierAccount",
        "duration_terms": "durationTerms",
        "estimated_days": "estimatedDays",
        "included_insurance_price": "includedInsurancePrice",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "provider_image_75": "providerImage75",
        "provider_image_200": "providerImage200",
    });
}));
function rateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Rate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rate' from JSON`);
}
