// shippo/models/components/servicegroupaccountandservicelevel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceGroupAccountAndServiceLevel$outboundSchema = exports.ServiceGroupAccountAndServiceLevel$inboundSchema = void 0;
exports.serviceGroupAccountAndServiceLevelToJSON = serviceGroupAccountAndServiceLevelToJSON;
exports.serviceGroupAccountAndServiceLevelFromJSON = serviceGroupAccountAndServiceLevelFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ServiceGroupAccountAndServiceLevel$inboundSchema = z.pipe(z.object({
    account_object_id: z.optional(z.string()),
    service_level_token: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_object_id": "accountObjectId",
        "service_level_token": "serviceLevelToken",
    });
}));
/** @internal */
exports.ServiceGroupAccountAndServiceLevel$outboundSchema = z.pipe(z.object({
    accountObjectId: z.optional(z.string()),
    serviceLevelToken: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountObjectId: "account_object_id",
        serviceLevelToken: "service_level_token",
    });
}));
function serviceGroupAccountAndServiceLevelToJSON(serviceGroupAccountAndServiceLevel) {
    return JSON.stringify(exports.ServiceGroupAccountAndServiceLevel$outboundSchema.parse(serviceGroupAccountAndServiceLevel));
}
function serviceGroupAccountAndServiceLevelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ServiceGroupAccountAndServiceLevel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ServiceGroupAccountAndServiceLevel' from JSON`);
}
