// shippo/models/components/servicelevelaustraliapostenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelAustraliaPostEnum$outboundSchema = exports.ServiceLevelAustraliaPostEnum$inboundSchema = exports.ServiceLevelAustraliaPostEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | australia_post_express_post | Express Post|
 * | australia_post_parcel_post | Parcel Post|
 * | australia_post_pack_and_track_international | Pack and Track International|
 * | australia_post_international_airmail | International Airmail|
 * | australia_post_express_post_international | Express Post International|
 * | australia_post_express_courier_international | Express Courier International|
 * | australia_post_international_express | International Express|
 * | australia_post_international_standard | International Standard|
 * | australia_post_international_economy | International Economy|
 * | australia_post_parcel_post_return | Parcel Post Return|
 */
exports.ServiceLevelAustraliaPostEnum = {
    AustraliaPostExpressPost: "australia_post_express_post",
    AustraliaPostParcelPost: "australia_post_parcel_post",
    AustraliaPostPackAndTrackInternational: "australia_post_pack_and_track_international",
    AustraliaPostInternationalAirmail: "australia_post_international_airmail",
    AustraliaPostExpressPostInternational: "australia_post_express_post_international",
    AustraliaPostExpressCourierInternational: "australia_post_express_courier_international",
    AustraliaPostInternationalExpress: "australia_post_international_express",
    AustraliaPostInternationalStandard: "australia_post_international_standard",
    AustraliaPostInternationalEconomy: "australia_post_international_economy",
    AustraliaPostParcelPostReturn: "australia_post_parcel_post_return",
};
/** @internal */
exports.ServiceLevelAustraliaPostEnum$inboundSchema = z.enum(exports.ServiceLevelAustraliaPostEnum);
/** @internal */
exports.ServiceLevelAustraliaPostEnum$outboundSchema = exports.ServiceLevelAustraliaPostEnum$inboundSchema;
