// shippo/models/components/servicelevelglobegisticsenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelGlobegisticsEnum$outboundSchema = exports.ServiceLevelGlobegisticsEnum$inboundSchema = exports.ServiceLevelGlobegisticsEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | globegistics_priority_mail_express_international | Globegistics Priority Mail Express International|
 * | globegistics_priority_mail_international | Globegistics Priority Mail International|
 * | globegistics_priority_mail_express_international_pds | Globegistics Priority Mail Express International PreSort Drop Ship|
 * | globegistics_priority_mail_international_pds | Globegistics Priority Mail International PreSort Drop Ship|
 * | globegistics_epacket | Globegistics ePacket|
 * | globegistics_ecom_tracked_ddp | Globegistics eCom Tracked DDP|
 * | globegistics_ecom_packet_ddp | Globegistics eCom Packet DDP|
 * | globegistics_ecom_priority_mail_international_ddp | Globegistics eCom Priority Mail International DDP|
 * | globegistics_ecom_priority_mail_express_international_ddp | Globegistics eCom Priority Mail Express International DDP|
 * | globegistics_ecom_extra | Globegistics eCom Extra|
 * | globegistics_ecom_international_priority_airmail | Globegistics eCom International Priority Airmail|
 * | globegistics_ecom_international_surface_airlift | Globegistics eCom International Surface Air Lift|
 */
exports.ServiceLevelGlobegisticsEnum = {
    GlobegisticsPriorityMailExpressInternational: "globegistics_priority_mail_express_international",
    GlobegisticsPriorityMailInternational: "globegistics_priority_mail_international",
    GlobegisticsPriorityMailExpressInternationalPds: "globegistics_priority_mail_express_international_pds",
    GlobegisticsPriorityMailInternationalPds: "globegistics_priority_mail_international_pds",
    GlobegisticsEpacket: "globegistics_epacket",
    GlobegisticsEcomTrackedDdp: "globegistics_ecom_tracked_ddp",
    GlobegisticsEcomPacketDdp: "globegistics_ecom_packet_ddp",
    GlobegisticsEcomPriorityMailInternationalDdp: "globegistics_ecom_priority_mail_international_ddp",
    GlobegisticsEcomPriorityMailExpressInternationalDdp: "globegistics_ecom_priority_mail_express_international_ddp",
    GlobegisticsEcomExtra: "globegistics_ecom_extra",
    GlobegisticsEcomInternationalPriorityAirmail: "globegistics_ecom_international_priority_airmail",
    GlobegisticsEcomInternationalSurfaceAirlift: "globegistics_ecom_international_surface_airlift",
};
/** @internal */
exports.ServiceLevelGlobegisticsEnum$inboundSchema = z.enum(exports.ServiceLevelGlobegisticsEnum);
/** @internal */
exports.ServiceLevelGlobegisticsEnum$outboundSchema = exports.ServiceLevelGlobegisticsEnum$inboundSchema;
