// shippo/models/components/servicelevelapcpostalenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelAPCPostalEnum$outboundSchema = exports.ServiceLevelAPCPostalEnum$inboundSchema = exports.ServiceLevelAPCPostalEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | apc_postal_parcelconnect_expedited | parcelConnect Expedited|
 * | apc_postal_parcelconnect_priority | parcelConnect Priority|
 * | apc_postal_parcelconnect_priority_delcon | parcelConnect Priority Delcon|
 * | apc_postal_parcelconnect_priority_pqw | parcelConnect Priority PQW|
 * | apc_postal_parcelconnect_book_service | parcelConnect Book Service|
 * | apc_postal_parcelconnect_standard | parcelConnect Standard|
 * | apc_postal_parcelconnect_epmi | parcelConnect ePMI|
 * | apc_postal_parcelconnect_epacket | parcelConnect ePacket|
 * | apc_postal_parcelconnect_epmei | parcelConnect ePMEI|
 */
exports.ServiceLevelAPCPostalEnum = {
    ApcPostalParcelconnectExpedited: "apc_postal_parcelconnect_expedited",
    ApcPostalParcelconnectPriority: "apc_postal_parcelconnect_priority",
    ApcPostalParcelconnectPriorityDelcon: "apc_postal_parcelconnect_priority_delcon",
    ApcPostalParcelconnectPriorityPqw: "apc_postal_parcelconnect_priority_pqw",
    ApcPostalParcelconnectBookService: "apc_postal_parcelconnect_book_service",
    ApcPostalParcelconnectStandard: "apc_postal_parcelconnect_standard",
    ApcPostalParcelconnectEpmi: "apc_postal_parcelconnect_epmi",
    ApcPostalParcelconnectEpacket: "apc_postal_parcelconnect_epacket",
    ApcPostalParcelconnectEpmei: "apc_postal_parcelconnect_epmei",
};
/** @internal */
exports.ServiceLevelAPCPostalEnum$inboundSchema = z.enum(exports.ServiceLevelAPCPostalEnum);
/** @internal */
exports.ServiceLevelAPCPostalEnum$outboundSchema = exports.ServiceLevelAPCPostalEnum$inboundSchema;
