// shippo/models/components/servicelevelglsusenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelGLSUSEnum$outboundSchema = exports.ServiceLevelGLSUSEnum$inboundSchema = exports.ServiceLevelGLSUSEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | gls_us_cps | GSO Ground|
 * | gls_us_eps | Early Priority Overnight|
 * | gls_us_ess | Early Saturday Delivery|
 * | gls_us_nps | Noon Priority Overnight|
 * | gls_us_pds | Priority Overnight|
 * | gls_us_sds | Saturday Delivery|
 */
exports.ServiceLevelGLSUSEnum = {
    GlsUsCps: "gls_us_cps",
    GlsUsEps: "gls_us_eps",
    GlsUsEss: "gls_us_ess",
    GlsUsNps: "gls_us_nps",
    GlsUsPds: "gls_us_pds",
    GlsUsSds: "gls_us_sds",
};
/** @internal */
exports.ServiceLevelGLSUSEnum$inboundSchema = z.enum(exports.ServiceLevelGLSUSEnum);
/** @internal */
exports.ServiceLevelGLSUSEnum$outboundSchema = exports.ServiceLevelGLSUSEnum$inboundSchema;
