// shippo/models/components/serviceleveluspsenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelUSPSEnum$outboundSchema = exports.ServiceLevelUSPSEnum$inboundSchema = exports.ServiceLevelUSPSEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | usps_priority | Priority Mail|
 * | usps_priority_express | Priority Mail Express|
 * | usps_media_mail | Media Mail, only for existing Shippo customers with grandfathered Media Mail option.|
 * | usps_priority_mail_international | Priority Mail International|
 * | usps_priority_mail_express_international | Priority Mail Express International|
 * | usps_first_class_package_international_service | First Class Package International|
 * | usps_ground_advantage | Ground Advantage |
 */
exports.ServiceLevelUSPSEnum = {
    UspsPriority: "usps_priority",
    UspsPriorityExpress: "usps_priority_express",
    UspsMediaMail: "usps_media_mail",
    UspsPriorityMailInternational: "usps_priority_mail_international",
    UspsPriorityMailExpressInternational: "usps_priority_mail_express_international",
    UspsFirstClassPackageInternationalService: "usps_first_class_package_international_service",
    UspsGroundAdvantage: "usps_ground_advantage",
};
/** @internal */
exports.ServiceLevelUSPSEnum$inboundSchema = z.enum(exports.ServiceLevelUSPSEnum);
/** @internal */
exports.ServiceLevelUSPSEnum$outboundSchema = exports.ServiceLevelUSPSEnum$inboundSchema;
