// shippo/models/components/servicegroupcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceGroupCreateRequest$outboundSchema = void 0;
exports.serviceGroupCreateRequestToJSON = serviceGroupCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const servicegroupaccountandservicelevel_js_1 = require("./servicegroupaccountandservicelevel.js");
const servicegrouptypeenum_js_1 = require("./servicegrouptypeenum.js");
/** @internal */
exports.ServiceGroupCreateRequest$outboundSchema = z.pipe(z.object({
    description: z.string(),
    flatRate: z.optional(z.string()),
    flatRateCurrency: z.optional(z.string()),
    freeShippingThresholdCurrency: z.optional(z.string()),
    freeShippingThresholdMin: z.optional(z.string()),
    name: z.string(),
    rateAdjustment: z.optional(z.int()),
    type: servicegrouptypeenum_js_1.ServiceGroupTypeEnum$outboundSchema,
    serviceLevels: z.array(servicegroupaccountandservicelevel_js_1.ServiceGroupAccountAndServiceLevel$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        flatRate: "flat_rate",
        flatRateCurrency: "flat_rate_currency",
        freeShippingThresholdCurrency: "free_shipping_threshold_currency",
        freeShippingThresholdMin: "free_shipping_threshold_min",
        rateAdjustment: "rate_adjustment",
        serviceLevels: "service_levels",
    });
}));
function serviceGroupCreateRequestToJSON(serviceGroupCreateRequest) {
    return JSON.stringify(exports.ServiceGroupCreateRequest$outboundSchema.parse(serviceGroupCreateRequest));
}
