// shippo/models/components/servicelevelasendiaenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelAsendiaEnum$outboundSchema = exports.ServiceLevelAsendiaEnum$inboundSchema = exports.ServiceLevelAsendiaEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | asendia_us_priority_tracked | Asendia USA Priority Tracked|
 * | asendia_us_international_express | Asendia USA International Express|
 * | asendia_us_international_priority_airmail | Asendia USA International Priority Airmail|
 * | asendia_us_international_surface_airlift | Asendia USA International Surface Air Lift|
 * | asendia_us_priority_mail_international | Asendia USA Priority Mail International|
 * | asendia_us_priority_mail_express_international | Asendia USA Priority Mail Express International|
 * | asendia_us_epacket | Asendia USA International ePacket|
 * | asendia_us_other | Asendia USA Other Services (custom)|
 */
exports.ServiceLevelAsendiaEnum = {
    AsendiaUsPriorityTracked: "asendia_us_priority_tracked",
    AsendiaUsInternationalExpress: "asendia_us_international_express",
    AsendiaUsInternationalPriorityAirmail: "asendia_us_international_priority_airmail",
    AsendiaUsInternationalSurfaceAirlift: "asendia_us_international_surface_airlift",
    AsendiaUsPriorityMailInternational: "asendia_us_priority_mail_international",
    AsendiaUsPriorityMailExpressInternational: "asendia_us_priority_mail_express_international",
    AsendiaUsEpacket: "asendia_us_epacket",
    AsendiaUsOther: "asendia_us_other",
};
/** @internal */
exports.ServiceLevelAsendiaEnum$inboundSchema = z.enum(exports.ServiceLevelAsendiaEnum);
/** @internal */
exports.ServiceLevelAsendiaEnum$outboundSchema = exports.ServiceLevelAsendiaEnum$inboundSchema;
