// shippo/models/components/servicelevelroyalmailsfenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelRoyalMailSFEnum$outboundSchema = exports.ServiceLevelRoyalMailSFEnum$inboundSchema = exports.ServiceLevelRoyalMailSFEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | royal_mail_sf_tracked_48 | Royal Mail Tracked 48|
 * | royal_mail_sf_tracked_48_for_post_office_drop | Royal Mail Tracked 48 for Post Office Drop|
 * | royal_mail_sf_international_tracked | Royal Mail International Tracked |
 * | royal_mail_sf_tracked_48_with_signature_confirmation | Royal Mail Tracked 48 with Signature |
 * | royal_mail_sf_tracked_48_for_post_office_drop_with_signature_confirmation | Royal Mail Tracked 48 for Post Office Drop with Signature |
 * | royal_mail_sf_tracked_24 | Royal Mail Tracked 24 |
 * | royal_mail_sf_tracked_24_for_post_office_drop | Royal Mail Tracked 24 for Post Office Drop |
 * | royal_mail_sf_tracked_24_for_post_office_drop_with_signature_confirmation | Royal Mail Tracked 24 for Post Office Drop with Signature |
 * | royal_mail_sf_international_tracked_for_post_office_drop | Royal Mail International Tracked for Post Office Drop |
 * | royal_mail_sf_international_tracked_with_signature_confirmation | Royal Mail International Tracked with Signature |
 * | royal_mail_sf_international_tracked_for_post_office_drop_with_signature_confirmation | Royal Mail International Tracked for Post Office Drop with Signature |
 */
exports.ServiceLevelRoyalMailSFEnum = {
    RoyalMailSfTracked48: "royal_mail_sf_tracked_48",
    RoyalMailSfTracked48ForPostOfficeDrop: "royal_mail_sf_tracked_48_for_post_office_drop",
    RoyalMailSfInternationalTracked: "royal_mail_sf_international_tracked",
    RoyalMailSfTracked48WithSignatureConfirmation: "royal_mail_sf_tracked_48_with_signature_confirmation",
    RoyalMailSfTracked48ForPostOfficeDropWithSignatureConfirmation: "royal_mail_sf_tracked_48_for_post_office_drop_with_signature_confirmation",
    RoyalMailSfTracked24: "royal_mail_sf_tracked_24",
    RoyalMailSfTracked24ForPostOfficeDrop: "royal_mail_sf_tracked_24_for_post_office_drop",
    RoyalMailSfTracked24ForPostOfficeDropWithSignatureConfirmation: "royal_mail_sf_tracked_24_for_post_office_drop_with_signature_confirmation",
    RoyalMailSfInternationalTrackedForPostOfficeDrop: "royal_mail_sf_international_tracked_for_post_office_drop",
    RoyalMailSfInternationalTrackedWithSignatureConfirmation: "royal_mail_sf_international_tracked_with_signature_confirmation",
    RoyalMailSfInternationalTrackedForPostOfficeDropWithSignatureConfirmation: "royal_mail_sf_international_tracked_for_post_office_drop_with_signature_confirmation",
};
/** @internal */
exports.ServiceLevelRoyalMailSFEnum$inboundSchema = z.enum(exports.ServiceLevelRoyalMailSFEnum);
/** @internal */
exports.ServiceLevelRoyalMailSFEnum$outboundSchema = exports.ServiceLevelRoyalMailSFEnum$inboundSchema;
