// shippo/models/components/servicelevellsoenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelLSOEnum$outboundSchema = exports.ServiceLevelLSOEnum$inboundSchema = exports.ServiceLevelLSOEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | lso_ground | Ground|
 * | lso_economy_next_day | Economy Next Day|
 * | lso_saturday_delivery | Saturday Delivery|
 * | lso_2nd_day | 2nd Day|
 * | lso_priority_next_day | Priority Next Day|
 * | lso_early_overnight | Early Overnight|
 */
exports.ServiceLevelLSOEnum = {
    LsoGround: "lso_ground",
    LsoEconomyNextDay: "lso_economy_next_day",
    LsoSaturdayDelivery: "lso_saturday_delivery",
    Lso2ndDay: "lso_2nd_day",
    LsoPriorityNextDay: "lso_priority_next_day",
    LsoEarlyOvernight: "lso_early_overnight",
};
/** @internal */
exports.ServiceLevelLSOEnum$inboundSchema = z.enum(exports.ServiceLevelLSOEnum);
/** @internal */
exports.ServiceLevelLSOEnum$outboundSchema = exports.ServiceLevelLSOEnum$inboundSchema;
