// shippo/models/components/serviceleveldhlgermanyenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelDHLGermanyEnum$outboundSchema = exports.ServiceLevelDHLGermanyEnum$inboundSchema = exports.ServiceLevelDHLGermanyEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_germany_europaket | DHL Germany Europaket|
 * | dhl_germany_paket | DHL Germany Paket|
 * | dhl_germany_paket_connect | DHL Germany Paket Connect|
 * | dhl_germany_paket_international | DHL Germany Paket International|
 * | dhl_germany_paket_priority | DHL Germany Paket Priority|
 * | dhl_germany_paket_sameday | DHL Germany Paket Sameday|
 */
exports.ServiceLevelDHLGermanyEnum = {
    DhlGermanyEuropaket: "dhl_germany_europaket",
    DhlGermanyPaket: "dhl_germany_paket",
    DhlGermanyPaketConnect: "dhl_germany_paket_connect",
    DhlGermanyPaketInternational: "dhl_germany_paket_international",
    DhlGermanyPaketPriority: "dhl_germany_paket_priority",
    DhlGermanyPaketSameday: "dhl_germany_paket_sameday",
};
/** @internal */
exports.ServiceLevelDHLGermanyEnum$inboundSchema = z.enum(exports.ServiceLevelDHLGermanyEnum);
/** @internal */
exports.ServiceLevelDHLGermanyEnum$outboundSchema = exports.ServiceLevelDHLGermanyEnum$inboundSchema;
