// shippo/models/components/servicelevelepostglobalenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelePostGlobalEnum$outboundSchema = exports.ServiceLevelePostGlobalEnum$inboundSchema = exports.ServiceLevelePostGlobalEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | rr_donnelley_domestic_economy_parcel | Domestic Economy Parcel|
 * | rr_donnelley_domestic_priority_parcel | Domestic Priority Parcel |
 * | rr_donnelley_domestic_parcel_bpm | Domestic Parcel BPM|
 * | rr_donnelley_priority_domestic_priority_parcel_bpm | Domestic Priority Parcel BPM|
 * | rr_donnelley_priority_parcel_delcon | International Priority Parcel DelCon|
 * | rr_donnelley_priority_parcel_nondelcon | International Priority Parcel NonDelcon|
 * | rr_donnelley_economy_parcel | Economy Parcel Service |
 * | rr_donnelley_ipa | International Priority Airmail (IPA)|
 * | rr_donnelley_courier | International Courier|
 * | rr_donnelley_isal | International Surface Air Lift (ISAL)|
 * | rr_donnelley_epacket | e-Packet|
 * | rr_donnelley_pmi | Priority Mail International|
 * | rr_donnelley_emi | Express Mail International|
 */
exports.ServiceLevelePostGlobalEnum = {
    RrDonnelleyDomesticEconomyParcel: "rr_donnelley_domestic_economy_parcel",
    RrDonnelleyDomesticPriorityParcel: "rr_donnelley_domestic_priority_parcel",
    RrDonnelleyDomesticParcelBpm: "rr_donnelley_domestic_parcel_bpm",
    RrDonnelleyPriorityDomesticPriorityParcelBpm: "rr_donnelley_priority_domestic_priority_parcel_bpm",
    RrDonnelleyPriorityParcelDelcon: "rr_donnelley_priority_parcel_delcon",
    RrDonnelleyPriorityParcelNondelcon: "rr_donnelley_priority_parcel_nondelcon",
    RrDonnelleyEconomyParcel: "rr_donnelley_economy_parcel",
    RrDonnelleyIpa: "rr_donnelley_ipa",
    RrDonnelleyCourier: "rr_donnelley_courier",
    RrDonnelleyIsal: "rr_donnelley_isal",
    RrDonnelleyEpacket: "rr_donnelley_epacket",
    RrDonnelleyPmi: "rr_donnelley_pmi",
    RrDonnelleyEmi: "rr_donnelley_emi",
};
/** @internal */
exports.ServiceLevelePostGlobalEnum$inboundSchema = z.enum(exports.ServiceLevelePostGlobalEnum);
/** @internal */
exports.ServiceLevelePostGlobalEnum$outboundSchema = exports.ServiceLevelePostGlobalEnum$inboundSchema;
