// shippo/models/components/servicelevelchronopostenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelChronopostEnum$outboundSchema = exports.ServiceLevelChronopostEnum$inboundSchema = exports.ServiceLevelChronopostEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | chronopost_13 | Chrono 13|
 * | chronopost_10 | Chrono 10|
 * | chronopost_18| Chrono 18|
 * | chronopost_relais_fr | Chrono Point Relais|
 * | chronopost_classic | Chrono Classic International|
 * | chronopost_express | Chrono Express International|
 */
exports.ServiceLevelChronopostEnum = {
    Chronopost13: "chronopost_13",
    Chronopost10: "chronopost_10",
    Chronopost18: "chronopost_18",
    ChronopostRelaisFr: "chronopost_relais_fr",
    ChronopostClassic: "chronopost_classic",
    ChronopostExpress: "chronopost_express",
};
/** @internal */
exports.ServiceLevelChronopostEnum$inboundSchema = z.enum(exports.ServiceLevelChronopostEnum);
/** @internal */
exports.ServiceLevelChronopostEnum$outboundSchema = exports.ServiceLevelChronopostEnum$inboundSchema;
