// shippo/models/components/servicelevelfedexenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelFedExEnum$outboundSchema = exports.ServiceLevelFedExEnum$inboundSchema = exports.ServiceLevelFedExEnum = void 0;
const z = __importStar(require("zod/v4-mini"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | fedex_ground | FedEx Ground®|
 * | fedex_home_delivery | FedEx Home Delivery®|
 * | fedex_ground_economy | FedEx Ground® Economy|
 * | fedex_2_day | FedEx 2Day®|
 * | fedex_2_day_am | FedEx 2Day® A.M.|
 * | fedex_express_saver | FedEx Express Saver®|
 * | fedex_standard_overnight | FedEx Standard Overnight®|
 * | fedex_priority_overnight | FedEx Priority Overnight®|
 * | fedex_first_overnight | FedEx First Overnight®|
 * | fedex_international_economy | FedEx International Economy®|
 * | fedex_international_priority | FedEx International Priority®|
 * | fedex_international_first | FedEx International First®|
 * | fedex_europe_first_international_priority | FedEx International First®|
 * | fedex_international_connect_plus | FedEx International Connect Plus|
 * | fedex_first | FedEx First |
 * | fedex_priority | FedEx Priority |
 * | fedex_priority_express | FedEx Priority Express|
 * | fedex_economy_select | FedEx® Economy |
 * | fedex_regional_economy | FedEx® Regional Economy |
 * | fedex_first_overnight_extra_hours | First Overnight® EH |
 * | fedex_international_priority_express | International Priority® Express |
 * | fedex_next_day_mid_morning | Next Day Mid Morning |
 * | fedex_priority_overnight_extra_hours | Priority Overnight® EH |
 * | fedex_standard_overnight_extra_hours | Standard Overnight® EH |
 */
exports.ServiceLevelFedExEnum = {
    FedexGround: "fedex_ground",
    FedexHomeDelivery: "fedex_home_delivery",
    FedexSmartPost: "fedex_smart_post",
    FedexGroundEconomy: "fedex_ground_economy",
    Fedex2Day: "fedex_2_day",
    Fedex2DayAm: "fedex_2_day_am",
    FedexExpressSaver: "fedex_express_saver",
    FedexStandardOvernight: "fedex_standard_overnight",
    FedexPriorityOvernight: "fedex_priority_overnight",
    FedexFirstOvernight: "fedex_first_overnight",
    FedexInternationalEconomy: "fedex_international_economy",
    FedexInternationalPriority: "fedex_international_priority",
    FedexInternationalFirst: "fedex_international_first",
    FedexEuropeFirstInternationalPriority: "fedex_europe_first_international_priority",
    FedexInternationalConnectPlus: "fedex_international_connect_plus",
    FedexFirst: "fedex_first",
    FedexPriority: "fedex_priority",
    FedexPriorityExpress: "fedex_priority_express",
    FedexEconomySelect: "fedex_economy_select",
    FedexRegionalEconomy: "fedex_regional_economy",
    FedexFirstOvernightExtraHours: "fedex_first_overnight_extra_hours",
    FedexInternationalPriorityExpress: "fedex_international_priority_express",
    FedexNextDayMidMorning: "fedex_next_day_mid_morning",
    FedexPriorityOvernightExtraHours: "fedex_priority_overnight_extra_hours",
    FedexStandardOvernightExtraHours: "fedex_standard_overnight_extra_hours",
};
/** @internal */
exports.ServiceLevelFedExEnum$inboundSchema = z.enum(exports.ServiceLevelFedExEnum);
/** @internal */
exports.ServiceLevelFedExEnum$outboundSchema = exports.ServiceLevelFedExEnum$inboundSchema;
