// shippo/models/components/shipment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Shipment$inboundSchema = exports.ShipmentStatus$inboundSchema = exports.ShipmentStatus = void 0;
exports.shipmentFromJSON = shipmentFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const customsdeclaration_js_1 = require("./customsdeclaration.js");
const parcel_js_1 = require("./parcel.js");
const rate_js_1 = require("./rate.js");
const responsemessage_js_1 = require("./responsemessage.js");
const shipmentextra_js_1 = require("./shipmentextra.js");
/**
 * `Waiting` shipments have been successfully submitted but not yet been processed.
 *
 * @remarks
 * `Queued` shipments are currently being processed.
 * `Success` shipments have been processed successfully, meaning that rate generation has concluded.
 * `Error` does not occur currently and is reserved for future use.
 */
exports.ShipmentStatus = {
    Error: "ERROR",
    Queued: "QUEUED",
    Success: "SUCCESS",
    Waiting: "WAITING",
};
/** @internal */
exports.ShipmentStatus$inboundSchema = z.enum(exports.ShipmentStatus);
/** @internal */
exports.Shipment$inboundSchema = z.pipe(z.object({
    extra: z.optional(shipmentextra_js_1.ShipmentExtra$inboundSchema),
    metadata: z.string(),
    shipment_date: z.optional(z.string()),
    address_from: address_js_1.Address$inboundSchema,
    address_return: z.optional(address_js_1.Address$inboundSchema),
    address_to: address_js_1.Address$inboundSchema,
    carrier_accounts: z.array(z.string()),
    customs_declaration: z.optional(customsdeclaration_js_1.CustomsDeclaration$inboundSchema),
    messages: z.array(responsemessage_js_1.ResponseMessage$inboundSchema),
    object_created: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    object_id: z.string(),
    object_owner: z.string(),
    object_updated: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    parcels: z.array(parcel_js_1.Parcel$inboundSchema),
    rates: z.array(rate_js_1.Rate$inboundSchema),
    status: exports.ShipmentStatus$inboundSchema,
    test: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "shipment_date": "shipmentDate",
        "address_from": "addressFrom",
        "address_return": "addressReturn",
        "address_to": "addressTo",
        "carrier_accounts": "carrierAccounts",
        "customs_declaration": "customsDeclaration",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_updated": "objectUpdated",
    });
}));
function shipmentFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Shipment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Shipment' from JSON`);
}
