// shippo/models/components/shipmentcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShipmentCreateRequest$outboundSchema = exports.Parcels$outboundSchema = exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema = exports.AddressTo$outboundSchema = exports.AddressReturn$outboundSchema = exports.AddressFrom$outboundSchema = void 0;
exports.addressFromToJSON = addressFromToJSON;
exports.addressReturnToJSON = addressReturnToJSON;
exports.addressToToJSON = addressToToJSON;
exports.shipmentCreateRequestCustomsDeclarationToJSON = shipmentCreateRequestCustomsDeclarationToJSON;
exports.parcelsToJSON = parcelsToJSON;
exports.shipmentCreateRequestToJSON = shipmentCreateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const customsdeclarationcreaterequest_js_1 = require("./customsdeclarationcreaterequest.js");
const parcelcreatefromtemplaterequest_js_1 = require("./parcelcreatefromtemplaterequest.js");
const parcelcreaterequest_js_1 = require("./parcelcreaterequest.js");
const shipmentextra_js_1 = require("./shipmentextra.js");
/** @internal */
exports.AddressFrom$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
function addressFromToJSON(addressFrom) {
    return JSON.stringify(exports.AddressFrom$outboundSchema.parse(addressFrom));
}
/** @internal */
exports.AddressReturn$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
function addressReturnToJSON(addressReturn) {
    return JSON.stringify(exports.AddressReturn$outboundSchema.parse(addressReturn));
}
/** @internal */
exports.AddressTo$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
function addressToToJSON(addressTo) {
    return JSON.stringify(exports.AddressTo$outboundSchema.parse(addressTo));
}
/** @internal */
exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema = z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$outboundSchema, z.string()]);
function shipmentCreateRequestCustomsDeclarationToJSON(shipmentCreateRequestCustomsDeclaration) {
    return JSON.stringify(exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema.parse(shipmentCreateRequestCustomsDeclaration));
}
/** @internal */
exports.Parcels$outboundSchema = z.union([
    parcelcreaterequest_js_1.ParcelCreateRequest$outboundSchema,
    parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$outboundSchema,
    z.string(),
]);
function parcelsToJSON(parcels) {
    return JSON.stringify(exports.Parcels$outboundSchema.parse(parcels));
}
/** @internal */
exports.ShipmentCreateRequest$outboundSchema = z.pipe(z.object({
    extra: z.optional(shipmentextra_js_1.ShipmentExtra$outboundSchema),
    metadata: z.optional(z.string()),
    shipmentDate: z.optional(z.string()),
    addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    addressReturn: z.optional(z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()])),
    addressTo: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    customsDeclaration: z.optional(z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$outboundSchema, z.string()])),
    async: z.optional(z.boolean()),
    carrierAccounts: z.optional(z.array(z.string())),
    parcels: z.array(z.union([
        parcelcreaterequest_js_1.ParcelCreateRequest$outboundSchema,
        parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$outboundSchema,
        z.string(),
    ])),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shipmentDate: "shipment_date",
        addressFrom: "address_from",
        addressReturn: "address_return",
        addressTo: "address_to",
        customsDeclaration: "customs_declaration",
        carrierAccounts: "carrier_accounts",
    });
}));
function shipmentCreateRequestToJSON(shipmentCreateRequest) {
    return JSON.stringify(exports.ShipmentCreateRequest$outboundSchema.parse(shipmentCreateRequest));
}
