// shippo/models/components/shipmentextra.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShipmentExtra$outboundSchema = exports.ShipmentExtra$inboundSchema = exports.SignatureConfirmation$outboundSchema = exports.SignatureConfirmation$inboundSchema = exports.ReturnServiceType$outboundSchema = exports.ReturnServiceType$inboundSchema = exports.PreferredDeliveryTimeframe$outboundSchema = exports.PreferredDeliveryTimeframe$inboundSchema = exports.DangerousGoodsCode$outboundSchema = exports.DangerousGoodsCode$inboundSchema = exports.AncillaryEndorsement$outboundSchema = exports.AncillaryEndorsement$inboundSchema = exports.SignatureConfirmation = exports.PreferredDeliveryTimeframe = exports.DangerousGoodsCode = exports.AncillaryEndorsement = void 0;
exports.returnServiceTypeToJSON = returnServiceTypeToJSON;
exports.returnServiceTypeFromJSON = returnServiceTypeFromJSON;
exports.shipmentExtraToJSON = shipmentExtraToJSON;
exports.shipmentExtraFromJSON = shipmentExtraFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const alcohol_js_1 = require("./alcohol.js");
const billing_js_1 = require("./billing.js");
const cod_js_1 = require("./cod.js");
const customerreference_js_1 = require("./customerreference.js");
const dangerousgoodsobject_js_1 = require("./dangerousgoodsobject.js");
const departmentnumber_js_1 = require("./departmentnumber.js");
const dryice_js_1 = require("./dryice.js");
const insurance_js_1 = require("./insurance.js");
const invoicenumber_js_1 = require("./invoicenumber.js");
const ponumber_js_1 = require("./ponumber.js");
const rmanumber_js_1 = require("./rmanumber.js");
const shipmentextralasershipattributesenum_js_1 = require("./shipmentextralasershipattributesenum.js");
const shipmentextrareturnservicetypelasershipenum_js_1 = require("./shipmentextrareturnservicetypelasershipenum.js");
const shipmentextrareturnservicetypeupsenum_js_1 = require("./shipmentextrareturnservicetypeupsenum.js");
const upsreferencefields_js_1 = require("./upsreferencefields.js");
/**
 * Specify an ancillary service endorsement to provide the USPS with instructions on how to handle undeliverable-as-addressed pieces (DHL eCommerce only).
 */
exports.AncillaryEndorsement = {
    ForwardingServiceRequested: "FORWARDING_SERVICE_REQUESTED",
    ReturnServiceRequested: "RETURN_SERVICE_REQUESTED",
};
/**
 * Dangerous Goods Code (DHL eCommerce only). See <a href="https://api-legacy.dhlecs.com/docs/v2/appendix.html#dangerous-goods">Category Codes</a>
 */
exports.DangerousGoodsCode = {
    One: "01",
    Two: "02",
    Three: "03",
    Four: "04",
    Five: "05",
    Six: "06",
    Seven: "07",
    Eight: "08",
    Nine: "09",
};
/**
 * Required for DHL Germany Paket Sameday. Designates a desired timeframe for delivery. Format is `HHMMHHMM`
 */
exports.PreferredDeliveryTimeframe = {
    TenMillionOneThousandTwoHundred: "10001200",
    TwelveMillionOneThousandFourHundred: "12001400",
    FourteenMillionOneThousandSixHundred: "14001600",
    SixteenMillionOneThousandEightHundred: "16001800",
    EighteenMillionTwoThousand: "18002000",
    NineteenMillionTwoThousandOneHundred: "19002100",
};
/**
 * Request standard or adult signature confirmation. You can alternatively request Certified Mail (USPS only)
 *
 * @remarks
 * or Indirect signature (FedEx only) or Carrier Confirmation (Deutsche Post only).
 */
exports.SignatureConfirmation = {
    Standard: "STANDARD",
    Adult: "ADULT",
    Certified: "CERTIFIED",
    Indirect: "INDIRECT",
    CarrierConfirmation: "CARRIER_CONFIRMATION",
};
/** @internal */
exports.AncillaryEndorsement$inboundSchema = z.enum(exports.AncillaryEndorsement);
/** @internal */
exports.AncillaryEndorsement$outboundSchema = exports.AncillaryEndorsement$inboundSchema;
/** @internal */
exports.DangerousGoodsCode$inboundSchema = z.enum(exports.DangerousGoodsCode);
/** @internal */
exports.DangerousGoodsCode$outboundSchema = exports.DangerousGoodsCode$inboundSchema;
/** @internal */
exports.PreferredDeliveryTimeframe$inboundSchema = z.enum(exports.PreferredDeliveryTimeframe);
/** @internal */
exports.PreferredDeliveryTimeframe$outboundSchema = exports.PreferredDeliveryTimeframe$inboundSchema;
/** @internal */
exports.ReturnServiceType$inboundSchema = z.union([
    shipmentextrareturnservicetypeupsenum_js_1.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema,
    shipmentextrareturnservicetypelasershipenum_js_1.ShipmentExtraReturnServiceTypeLasershipEnum$inboundSchema,
]);
/** @internal */
exports.ReturnServiceType$outboundSchema = z.union([
    shipmentextrareturnservicetypeupsenum_js_1.ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema,
    shipmentextrareturnservicetypelasershipenum_js_1.ShipmentExtraReturnServiceTypeLasershipEnum$outboundSchema,
]);
function returnServiceTypeToJSON(returnServiceType) {
    return JSON.stringify(exports.ReturnServiceType$outboundSchema.parse(returnServiceType));
}
function returnServiceTypeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ReturnServiceType$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReturnServiceType' from JSON`);
}
/** @internal */
exports.SignatureConfirmation$inboundSchema = z.enum(exports.SignatureConfirmation);
/** @internal */
exports.SignatureConfirmation$outboundSchema = exports.SignatureConfirmation$inboundSchema;
/** @internal */
exports.ShipmentExtra$inboundSchema = z.pipe(z.object({
    accounts_receivable_customer_account: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    alcohol: z.optional(alcohol_js_1.Alcohol$inboundSchema),
    ancillary_endorsement: z.optional(exports.AncillaryEndorsement$inboundSchema),
    appropriation_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    authority_to_leave: z.optional(z.boolean()),
    bill_of_lading_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    billing: z.optional(billing_js_1.Billing$inboundSchema),
    bypass_address_validation: z.optional(z.boolean()),
    carbon_neutral: z.optional(z.boolean()),
    carrier_hub_id: z.optional(z.string()),
    carrier_hub_travel_time: z.optional(z.int()),
    COD: z.optional(cod_js_1.Cod$inboundSchema),
    cod_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    container_type: z.optional(z.string()),
    critical_pull_time: z.optional(z.string()),
    customer_branch: z.optional(z.string()),
    customer_reference: z.optional(customerreference_js_1.CustomerReference$inboundSchema),
    dangerous_goods: z.optional(dangerousgoodsobject_js_1.DangerousGoodsObject$inboundSchema),
    dangerous_goods_code: z.optional(exports.DangerousGoodsCode$inboundSchema),
    dealer_order_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    delivery_instructions: z.optional(z.string()),
    dept_number: z.optional(departmentnumber_js_1.DepartmentNumber$inboundSchema),
    dry_ice: z.optional(dryice_js_1.DryIce$inboundSchema),
    fda_product_code: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    fulfillment_center: z.optional(z.string()),
    insurance: z.optional(insurance_js_1.Insurance$inboundSchema),
    invoice_number: z.optional(invoicenumber_js_1.InvoiceNumber$inboundSchema),
    is_return: z.optional(z.boolean()),
    lasership_attrs: z.optional(z.array(shipmentextralasershipattributesenum_js_1.ShipmentExtraLasershipAttributesEnum$inboundSchema)),
    lasership_declared_value: z.optional(z.string()),
    manifest_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    model_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    part_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    po_number: z.optional(ponumber_js_1.PoNumber$inboundSchema),
    preferred_delivery_timeframe: z.optional(exports.PreferredDeliveryTimeframe$inboundSchema),
    premium: z.optional(z.boolean()),
    production_code: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    purchase_request_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    qr_code_requested: z.optional(z.boolean()),
    reference_1: z.optional(z.string()),
    reference_2: z.optional(z.string()),
    request_retail_rates: z.optional(z.boolean()),
    return_service_type: z.optional(z.union([
        shipmentextrareturnservicetypeupsenum_js_1.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema,
        shipmentextrareturnservicetypelasershipenum_js_1.ShipmentExtraReturnServiceTypeLasershipEnum$inboundSchema,
    ])),
    rma_number: z.optional(rmanumber_js_1.RmaNumber$inboundSchema),
    saturday_delivery: z.optional(z.boolean()),
    salesperson_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    serial_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    signature_confirmation: z.optional(exports.SignatureConfirmation$inboundSchema),
    store_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    transaction_reference_number: z.optional(upsreferencefields_js_1.UPSReferenceFields$inboundSchema),
    usmca_eligible: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "accounts_receivable_customer_account": "accountsReceivableCustomerAccount",
        "ancillary_endorsement": "ancillaryEndorsement",
        "appropriation_number": "appropriationNumber",
        "authority_to_leave": "authorityToLeave",
        "bill_of_lading_number": "billOfLadingNumber",
        "bypass_address_validation": "bypassAddressValidation",
        "carbon_neutral": "carbonNeutral",
        "carrier_hub_id": "carrierHubId",
        "carrier_hub_travel_time": "carrierHubTravelTime",
        "COD": "cod",
        "cod_number": "codNumber",
        "container_type": "containerType",
        "critical_pull_time": "criticalPullTime",
        "customer_branch": "customerBranch",
        "customer_reference": "customerReference",
        "dangerous_goods": "dangerousGoods",
        "dangerous_goods_code": "dangerousGoodsCode",
        "dealer_order_number": "dealerOrderNumber",
        "delivery_instructions": "deliveryInstructions",
        "dept_number": "deptNumber",
        "dry_ice": "dryIce",
        "fda_product_code": "fdaProductCode",
        "fulfillment_center": "fulfillmentCenter",
        "invoice_number": "invoiceNumber",
        "is_return": "isReturn",
        "lasership_attrs": "lasershipAttrs",
        "lasership_declared_value": "lasershipDeclaredValue",
        "manifest_number": "manifestNumber",
        "model_number": "modelNumber",
        "part_number": "partNumber",
        "po_number": "poNumber",
        "preferred_delivery_timeframe": "preferredDeliveryTimeframe",
        "production_code": "productionCode",
        "purchase_request_number": "purchaseRequestNumber",
        "qr_code_requested": "qrCodeRequested",
        "reference_1": "reference1",
        "reference_2": "reference2",
        "request_retail_rates": "requestRetailRates",
        "return_service_type": "returnServiceType",
        "rma_number": "rmaNumber",
        "saturday_delivery": "saturdayDelivery",
        "salesperson_number": "salespersonNumber",
        "serial_number": "serialNumber",
        "signature_confirmation": "signatureConfirmation",
        "store_number": "storeNumber",
        "transaction_reference_number": "transactionReferenceNumber",
        "usmca_eligible": "usmcaEligible",
    });
}));
/** @internal */
exports.ShipmentExtra$outboundSchema = z.pipe(z.object({
    accountsReceivableCustomerAccount: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    alcohol: z.optional(alcohol_js_1.Alcohol$outboundSchema),
    ancillaryEndorsement: z.optional(exports.AncillaryEndorsement$outboundSchema),
    appropriationNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    authorityToLeave: z.optional(z.boolean()),
    billOfLadingNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    billing: z.optional(billing_js_1.Billing$outboundSchema),
    bypassAddressValidation: z.optional(z.boolean()),
    carbonNeutral: z.optional(z.boolean()),
    carrierHubId: z.optional(z.string()),
    carrierHubTravelTime: z.optional(z.int()),
    cod: z.optional(cod_js_1.Cod$outboundSchema),
    codNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    containerType: z.optional(z.string()),
    criticalPullTime: z.optional(z.string()),
    customerBranch: z.optional(z.string()),
    customerReference: z.optional(customerreference_js_1.CustomerReference$outboundSchema),
    dangerousGoods: z.optional(dangerousgoodsobject_js_1.DangerousGoodsObject$outboundSchema),
    dangerousGoodsCode: z.optional(exports.DangerousGoodsCode$outboundSchema),
    dealerOrderNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    deliveryInstructions: z.optional(z.string()),
    deptNumber: z.optional(departmentnumber_js_1.DepartmentNumber$outboundSchema),
    dryIce: z.optional(dryice_js_1.DryIce$outboundSchema),
    fdaProductCode: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    fulfillmentCenter: z.optional(z.string()),
    insurance: z.optional(insurance_js_1.Insurance$outboundSchema),
    invoiceNumber: z.optional(invoicenumber_js_1.InvoiceNumber$outboundSchema),
    isReturn: z.optional(z.boolean()),
    lasershipAttrs: z.optional(z.array(shipmentextralasershipattributesenum_js_1.ShipmentExtraLasershipAttributesEnum$outboundSchema)),
    lasershipDeclaredValue: z.optional(z.string()),
    manifestNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    modelNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    partNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    poNumber: z.optional(ponumber_js_1.PoNumber$outboundSchema),
    preferredDeliveryTimeframe: z.optional(exports.PreferredDeliveryTimeframe$outboundSchema),
    premium: z.optional(z.boolean()),
    productionCode: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    purchaseRequestNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    qrCodeRequested: z.optional(z.boolean()),
    reference1: z.optional(z.string()),
    reference2: z.optional(z.string()),
    requestRetailRates: z.optional(z.boolean()),
    returnServiceType: z.optional(z.union([
        shipmentextrareturnservicetypeupsenum_js_1.ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema,
        shipmentextrareturnservicetypelasershipenum_js_1.ShipmentExtraReturnServiceTypeLasershipEnum$outboundSchema,
    ])),
    rmaNumber: z.optional(rmanumber_js_1.RmaNumber$outboundSchema),
    saturdayDelivery: z.optional(z.boolean()),
    salespersonNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    serialNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    signatureConfirmation: z.optional(exports.SignatureConfirmation$outboundSchema),
    storeNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    transactionReferenceNumber: z.optional(upsreferencefields_js_1.UPSReferenceFields$outboundSchema),
    usmcaEligible: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountsReceivableCustomerAccount: "accounts_receivable_customer_account",
        ancillaryEndorsement: "ancillary_endorsement",
        appropriationNumber: "appropriation_number",
        authorityToLeave: "authority_to_leave",
        billOfLadingNumber: "bill_of_lading_number",
        bypassAddressValidation: "bypass_address_validation",
        carbonNeutral: "carbon_neutral",
        carrierHubId: "carrier_hub_id",
        carrierHubTravelTime: "carrier_hub_travel_time",
        cod: "COD",
        codNumber: "cod_number",
        containerType: "container_type",
        criticalPullTime: "critical_pull_time",
        customerBranch: "customer_branch",
        customerReference: "customer_reference",
        dangerousGoods: "dangerous_goods",
        dangerousGoodsCode: "dangerous_goods_code",
        dealerOrderNumber: "dealer_order_number",
        deliveryInstructions: "delivery_instructions",
        deptNumber: "dept_number",
        dryIce: "dry_ice",
        fdaProductCode: "fda_product_code",
        fulfillmentCenter: "fulfillment_center",
        invoiceNumber: "invoice_number",
        isReturn: "is_return",
        lasershipAttrs: "lasership_attrs",
        lasershipDeclaredValue: "lasership_declared_value",
        manifestNumber: "manifest_number",
        modelNumber: "model_number",
        partNumber: "part_number",
        poNumber: "po_number",
        preferredDeliveryTimeframe: "preferred_delivery_timeframe",
        productionCode: "production_code",
        purchaseRequestNumber: "purchase_request_number",
        qrCodeRequested: "qr_code_requested",
        reference1: "reference_1",
        reference2: "reference_2",
        requestRetailRates: "request_retail_rates",
        returnServiceType: "return_service_type",
        rmaNumber: "rma_number",
        saturdayDelivery: "saturday_delivery",
        salespersonNumber: "salesperson_number",
        serialNumber: "serial_number",
        signatureConfirmation: "signature_confirmation",
        storeNumber: "store_number",
        transactionReferenceNumber: "transaction_reference_number",
        usmcaEligible: "usmca_eligible",
    });
}));
function shipmentExtraToJSON(shipmentExtra) {
    return JSON.stringify(exports.ShipmentExtra$outboundSchema.parse(shipmentExtra));
}
function shipmentExtraFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShipmentExtra$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShipmentExtra' from JSON`);
}
