// shippo/models/components/trackingstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackingStatus$outboundSchema = exports.TrackingStatus$inboundSchema = void 0;
exports.trackingStatusToJSON = trackingStatusToJSON;
exports.trackingStatusFromJSON = trackingStatusFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const trackingstatusenum_js_1 = require("./trackingstatusenum.js");
const trackingstatuslocationbase_js_1 = require("./trackingstatuslocationbase.js");
const trackingstatussubstatus_js_1 = require("./trackingstatussubstatus.js");
/** @internal */
exports.TrackingStatus$inboundSchema = z.pipe(z.object({
    location: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$inboundSchema),
    object_created: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    object_id: z.string(),
    object_updated: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    status: trackingstatusenum_js_1.TrackingStatusEnum$inboundSchema,
    substatus: z.optional(trackingstatussubstatus_js_1.TrackingStatusSubstatus$inboundSchema),
    status_date: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    status_details: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_updated": "objectUpdated",
        "status_date": "statusDate",
        "status_details": "statusDetails",
    });
}));
/** @internal */
exports.TrackingStatus$outboundSchema = z.pipe(z.object({
    location: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$outboundSchema),
    objectCreated: z.pipe(z.date(), z.transform(v => v.toISOString())),
    objectId: z.string(),
    objectUpdated: z.pipe(z.date(), z.transform(v => v.toISOString())),
    status: trackingstatusenum_js_1.TrackingStatusEnum$outboundSchema,
    substatus: z.optional(trackingstatussubstatus_js_1.TrackingStatusSubstatus$outboundSchema),
    statusDate: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    statusDetails: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        objectCreated: "object_created",
        objectId: "object_id",
        objectUpdated: "object_updated",
        statusDate: "status_date",
        statusDetails: "status_details",
    });
}));
function trackingStatusToJSON(trackingStatus) {
    return JSON.stringify(exports.TrackingStatus$outboundSchema.parse(trackingStatus));
}
function trackingStatusFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TrackingStatus$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TrackingStatus' from JSON`);
}
