// shippo/models/components/track.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Track$outboundSchema = exports.Track$inboundSchema = void 0;
exports.trackToJSON = trackToJSON;
exports.trackFromJSON = trackFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const servicelevelwithparent_js_1 = require("./servicelevelwithparent.js");
const trackingstatus_js_1 = require("./trackingstatus.js");
const trackingstatuslocationbase_js_1 = require("./trackingstatuslocationbase.js");
/** @internal */
exports.Track$inboundSchema = z.pipe(z.object({
    address_from: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$inboundSchema),
    address_to: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$inboundSchema),
    carrier: z.string(),
    eta: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    messages: z.array(z.string()),
    metadata: z.optional(z.string()),
    original_eta: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    servicelevel: z.optional(servicelevelwithparent_js_1.ServiceLevelWithParent$inboundSchema),
    tracking_history: z.array(trackingstatus_js_1.TrackingStatus$inboundSchema),
    tracking_number: z.string(),
    tracking_status: z.optional(trackingstatus_js_1.TrackingStatus$inboundSchema),
    transaction: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_from": "addressFrom",
        "address_to": "addressTo",
        "original_eta": "originalEta",
        "tracking_history": "trackingHistory",
        "tracking_number": "trackingNumber",
        "tracking_status": "trackingStatus",
    });
}));
/** @internal */
exports.Track$outboundSchema = z
    .pipe(z.object({
    addressFrom: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$outboundSchema),
    addressTo: z.optional(trackingstatuslocationbase_js_1.TrackingStatusLocationBase$outboundSchema),
    carrier: z.string(),
    eta: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    messages: z.array(z.string()),
    metadata: z.optional(z.string()),
    originalEta: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    servicelevel: z.optional(servicelevelwithparent_js_1.ServiceLevelWithParent$outboundSchema),
    trackingHistory: z.array(trackingstatus_js_1.TrackingStatus$outboundSchema),
    trackingNumber: z.string(),
    trackingStatus: z.optional(trackingstatus_js_1.TrackingStatus$outboundSchema),
    transaction: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressFrom: "address_from",
        addressTo: "address_to",
        originalEta: "original_eta",
        trackingHistory: "tracking_history",
        trackingNumber: "tracking_number",
        trackingStatus: "tracking_status",
    });
}));
function trackToJSON(track) {
    return JSON.stringify(exports.Track$outboundSchema.parse(track));
}
function trackFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Track$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Track' from JSON`);
}
