// shippo/models/components/transaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction$outboundSchema = exports.Transaction$inboundSchema = exports.TransactionRate$outboundSchema = exports.TransactionRate$inboundSchema = exports.CreatedBy$outboundSchema = exports.CreatedBy$inboundSchema = void 0;
exports.createdByToJSON = createdByToJSON;
exports.createdByFromJSON = createdByFromJSON;
exports.transactionRateToJSON = transactionRateToJSON;
exports.transactionRateFromJSON = transactionRateFromJSON;
exports.transactionToJSON = transactionToJSON;
exports.transactionFromJSON = transactionFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const corerate_js_1 = require("./corerate.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
const responsemessage_js_1 = require("./responsemessage.js");
const trackingstatusenum_js_1 = require("./trackingstatusenum.js");
const transactionstatusenum_js_1 = require("./transactionstatusenum.js");
/** @internal */
exports.CreatedBy$inboundSchema = z
    .pipe(z.object({
    first_name: z.optional(z.string()),
    last_name: z.optional(z.string()),
    username: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "first_name": "firstName",
        "last_name": "lastName",
    });
}));
/** @internal */
exports.CreatedBy$outboundSchema = z.pipe(z.object({
    firstName: z.optional(z.string()),
    lastName: z.optional(z.string()),
    username: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        firstName: "first_name",
        lastName: "last_name",
    });
}));
function createdByToJSON(createdBy) {
    return JSON.stringify(exports.CreatedBy$outboundSchema.parse(createdBy));
}
function createdByFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreatedBy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreatedBy' from JSON`);
}
/** @internal */
exports.TransactionRate$inboundSchema = z.union([corerate_js_1.CoreRate$inboundSchema, z.string()]);
/** @internal */
exports.TransactionRate$outboundSchema = z.union([corerate_js_1.CoreRate$outboundSchema, z.string()]);
function transactionRateToJSON(transactionRate) {
    return JSON.stringify(exports.TransactionRate$outboundSchema.parse(transactionRate));
}
function transactionRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TransactionRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransactionRate' from JSON`);
}
/** @internal */
exports.Transaction$inboundSchema = z
    .pipe(z.object({
    commercial_invoice_url: z.optional(z.string()),
    created_by: z.optional(z.nullable(z.lazy(() => exports.CreatedBy$inboundSchema))),
    eta: z.optional(z.string()),
    label_file_type: z.optional(labelfiletypeenum_js_1.LabelFileTypeEnum$inboundSchema),
    label_url: z.optional(z.string()),
    messages: z.optional(z.array(responsemessage_js_1.ResponseMessage$inboundSchema)),
    metadata: z.optional(z.string()),
    object_created: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    object_id: z.optional(z.string()),
    object_owner: z.optional(z.string()),
    object_state: z.optional(objectstateenum_js_1.ObjectStateEnum$inboundSchema),
    object_updated: z.optional(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    parcel: z.optional(z.string()),
    qr_code_url: z.optional(z.string()),
    rate: z.optional(z.union([corerate_js_1.CoreRate$inboundSchema, z.string()])),
    status: z.optional(transactionstatusenum_js_1.TransactionStatusEnum$inboundSchema),
    test: z.optional(z.boolean()),
    tracking_number: z.optional(z.string()),
    tracking_status: z.optional(trackingstatusenum_js_1.TrackingStatusEnum$inboundSchema),
    tracking_url_provider: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "commercial_invoice_url": "commercialInvoiceUrl",
        "created_by": "createdBy",
        "label_file_type": "labelFileType",
        "label_url": "labelUrl",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_state": "objectState",
        "object_updated": "objectUpdated",
        "qr_code_url": "qrCodeUrl",
        "tracking_number": "trackingNumber",
        "tracking_status": "trackingStatus",
        "tracking_url_provider": "trackingUrlProvider",
    });
}));
/** @internal */
exports.Transaction$outboundSchema = z.pipe(z.object({
    commercialInvoiceUrl: z.optional(z.string()),
    createdBy: z.optional(z.nullable(z.lazy(() => exports.CreatedBy$outboundSchema))),
    eta: z.optional(z.string()),
    labelFileType: z.optional(labelfiletypeenum_js_1.LabelFileTypeEnum$outboundSchema),
    labelUrl: z.optional(z.string()),
    messages: z.optional(z.array(responsemessage_js_1.ResponseMessage$outboundSchema)),
    metadata: z.optional(z.string()),
    objectCreated: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    objectId: z.optional(z.string()),
    objectOwner: z.optional(z.string()),
    objectState: z.optional(objectstateenum_js_1.ObjectStateEnum$outboundSchema),
    objectUpdated: z.optional(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    parcel: z.optional(z.string()),
    qrCodeUrl: z.optional(z.string()),
    rate: z.optional(z.union([corerate_js_1.CoreRate$outboundSchema, z.string()])),
    status: z.optional(transactionstatusenum_js_1.TransactionStatusEnum$outboundSchema),
    test: z.optional(z.boolean()),
    trackingNumber: z.optional(z.string()),
    trackingStatus: z.optional(trackingstatusenum_js_1.TrackingStatusEnum$outboundSchema),
    trackingUrlProvider: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        commercialInvoiceUrl: "commercial_invoice_url",
        createdBy: "created_by",
        labelFileType: "label_file_type",
        labelUrl: "label_url",
        objectCreated: "object_created",
        objectId: "object_id",
        objectOwner: "object_owner",
        objectState: "object_state",
        objectUpdated: "object_updated",
        qrCodeUrl: "qr_code_url",
        trackingNumber: "tracking_number",
        trackingStatus: "tracking_status",
        trackingUrlProvider: "tracking_url_provider",
    });
}));
function transactionToJSON(transaction) {
    return JSON.stringify(exports.Transaction$outboundSchema.parse(transaction));
}
function transactionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Transaction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Transaction' from JSON`);
}
