// shippo/models/components/webhookpayloadtrack.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookPayloadTrack$outboundSchema = exports.WebhookPayloadTrack$inboundSchema = void 0;
exports.webhookPayloadTrackToJSON = webhookPayloadTrackToJSON;
exports.webhookPayloadTrackFromJSON = webhookPayloadTrackFromJSON;
const z = __importStar(require("zod/v4-mini"));
const schemas_js_1 = require("../../lib/schemas.js");
const track_js_1 = require("./track.js");
const webhookeventtypeenum_js_1 = require("./webhookeventtypeenum.js");
/** @internal */
exports.WebhookPayloadTrack$inboundSchema = z.object({
    event: z.optional(webhookeventtypeenum_js_1.WebhookEventTypeEnum$inboundSchema),
    test: z.optional(z.boolean()),
    data: z.optional(track_js_1.Track$inboundSchema),
});
/** @internal */
exports.WebhookPayloadTrack$outboundSchema = z.object({
    event: z.optional(webhookeventtypeenum_js_1.WebhookEventTypeEnum$outboundSchema),
    test: z.optional(z.boolean()),
    data: z.optional(track_js_1.Track$outboundSchema),
});
function webhookPayloadTrackToJSON(webhookPayloadTrack) {
    return JSON.stringify(exports.WebhookPayloadTrack$outboundSchema.parse(webhookPayloadTrack));
}
function webhookPayloadTrackFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPayloadTrack$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPayloadTrack' from JSON`);
}
