// shippo/models/operations/initiateoauth2signin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InitiateOauth2SigninResponse$inboundSchema = exports.InitiateOauth2SigninRequest$outboundSchema = void 0;
exports.initiateOauth2SigninRequestToJSON = initiateOauth2SigninRequestToJSON;
exports.initiateOauth2SigninResponseFromJSON = initiateOauth2SigninResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.InitiateOauth2SigninRequest$outboundSchema = z.pipe(z.object({
    carrierAccountObjectId: z.string(),
    redirectUri: z.string(),
    state: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccountObjectId: "CarrierAccountObjectId",
        redirectUri: "redirect_uri",
    });
}));
function initiateOauth2SigninRequestToJSON(initiateOauth2SigninRequest) {
    return JSON.stringify(exports.InitiateOauth2SigninRequest$outboundSchema.parse(initiateOauth2SigninRequest));
}
/** @internal */
exports.InitiateOauth2SigninResponse$inboundSchema = z.pipe(z.object({
    Headers: z._default(z.record(z.string(), z.array(z.string())), {}),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Headers": "headers",
    });
}));
function initiateOauth2SigninResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.InitiateOauth2SigninResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InitiateOauth2SigninResponse' from JSON`);
}
