// shippo/models/operations/listshipments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListShipmentsRequest$outboundSchema = void 0;
exports.listShipmentsRequestToJSON = listShipmentsRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.ListShipmentsRequest$outboundSchema = z.pipe(z.object({
    pageToken: z.optional(z.string()),
    page: z._default(z.int(), 1),
    results: z._default(z.int(), 25),
    objectCreatedGt: z.optional(z.string()),
    objectCreatedGte: z.optional(z.string()),
    objectCreatedLt: z.optional(z.string()),
    objectCreatedLte: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageToken: "page_token",
        objectCreatedGt: "object_created_gt",
        objectCreatedGte: "object_created_gte",
        objectCreatedLt: "object_created_lt",
        objectCreatedLte: "object_created_lte",
    });
}));
function listShipmentsRequestToJSON(listShipmentsRequest) {
    return JSON.stringify(exports.ListShipmentsRequest$outboundSchema.parse(listShipmentsRequest));
}
