// shippo/models/errors/initiateoauth2signin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InitiateOauth2SigninResponseBody$inboundSchema = exports.InitiateOauth2SigninCarrierAccountsResponseBody$inboundSchema = exports.InitiateOauth2SigninCarrierAccountsResponseResponseBody$inboundSchema = exports.InitiateOauth2SigninResponseBody = exports.InitiateOauth2SigninCarrierAccountsResponseBody = exports.InitiateOauth2SigninCarrierAccountsResponseResponseBody = void 0;
const z = __importStar(require("zod/v4-mini"));
const shippoerror_js_1 = require("./shippoerror.js");
/**
 * Invalid carrier account provided by the user
 */
class InitiateOauth2SigninCarrierAccountsResponseResponseBody extends shippoerror_js_1.ShippoError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        if (err.title != null)
            this.title = err.title;
        if (err.detail != null)
            this.detail = err.detail;
        this.name = "InitiateOauth2SigninCarrierAccountsResponseResponseBody";
    }
}
exports.InitiateOauth2SigninCarrierAccountsResponseResponseBody = InitiateOauth2SigninCarrierAccountsResponseResponseBody;
/**
 * Invalid ShippoToken or unsupported carrier account provided by the user
 */
class InitiateOauth2SigninCarrierAccountsResponseBody extends shippoerror_js_1.ShippoError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        if (err.title != null)
            this.title = err.title;
        if (err.detail != null)
            this.detail = err.detail;
        this.name = "InitiateOauth2SigninCarrierAccountsResponseBody";
    }
}
exports.InitiateOauth2SigninCarrierAccountsResponseBody = InitiateOauth2SigninCarrierAccountsResponseBody;
/**
 * Invalid parameters provided by the user
 */
class InitiateOauth2SigninResponseBody extends shippoerror_js_1.ShippoError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        if (err.title != null)
            this.title = err.title;
        if (err.detail != null)
            this.detail = err.detail;
        this.name = "InitiateOauth2SigninResponseBody";
    }
}
exports.InitiateOauth2SigninResponseBody = InitiateOauth2SigninResponseBody;
/** @internal */
exports.InitiateOauth2SigninCarrierAccountsResponseResponseBody$inboundSchema = z.pipe(z.object({
    title: z.optional(z.string()),
    detail: z.optional(z.string()),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
}), z.transform((v) => {
    return new InitiateOauth2SigninCarrierAccountsResponseResponseBody(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
}));
/** @internal */
exports.InitiateOauth2SigninCarrierAccountsResponseBody$inboundSchema = z
    .pipe(z.object({
    title: z.optional(z.string()),
    detail: z.optional(z.string()),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
}), z.transform((v) => {
    return new InitiateOauth2SigninCarrierAccountsResponseBody(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
}));
/** @internal */
exports.InitiateOauth2SigninResponseBody$inboundSchema = z.pipe(z.object({
    title: z.optional(z.string()),
    detail: z.optional(z.string()),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
}), z.transform((v) => {
    return new InitiateOauth2SigninResponseBody(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
}));
